
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://iliapopov17.github.io/NGS-Handbook/03_Whole_%28pan%29genome_analyses/">
      
      
        <link rel="prev" href="../02_Variant_Calling/">
      
      
        <link rel="next" href="../IV.%20Phylogenetics/04_00_Overview/">
      
      
      <link rel="icon" href="../assets/favicon.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.49">
    
    
      
        <title>III. Whole Genome and Pangenome Analyses - NGS Analysis Handbook</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.6f8fc17f.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  

<script id="__analytics">function __md_analytics(){function e(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],e("js",new Date),e("config","G-4RBMQR50GK"),document.addEventListener("DOMContentLoaded",(function(){document.forms.search&&document.forms.search.query.addEventListener("blur",(function(){this.value&&e("event","search",{search_term:this.value})}));document$.subscribe((function(){var t=document.forms.feedback;if(void 0!==t)for(var a of t.querySelectorAll("[type=submit]"))a.addEventListener("click",(function(a){a.preventDefault();var n=document.location.pathname,d=this.getAttribute("data-md-value");e("event","feedback",{page:n,data:d}),t.firstElementChild.disabled=!0;var r=t.querySelector(".md-feedback__note [data-md-value='"+d+"']");r&&(r.hidden=!1)})),t.hidden=!1})),location$.subscribe((function(t){e("config","G-4RBMQR50GK",{page_path:t.pathname})}))}));var t=document.createElement("script");t.async=!0,t.src="https://www.googletagmanager.com/gtag/js?id=G-4RBMQR50GK",document.getElementById("__analytics").insertAdjacentElement("afterEnd",t)}</script>
  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
    
    <html data-lt-installed="true">
        <head>
            <!-- HTML Meta Tags -->
            <title>NGS Data Analysis Handbook</title>
            <meta name="description" content="">

            <!-- Facebook Meta Tags -->
            <meta property="og:url" content="https://iliapopov17.github.io/NGS-Handbook/">
            <meta property="og:type" content="website">
            <meta property="og:title" content="NGS Data Analysis Handbook">
            <meta property="og:description" content="">
            <meta property="og:image" content="https://iliapopov17.github.io/images/ngs-handbook/metatag.png">

            <!-- Twitter Meta Tags -->
            <meta name="twitter:card" content="summary_large_image">
            <meta property="twitter:domain" content="iliapopov17.github.io">
            <meta property="twitter:url" content="https://iliapopov17.github.io/NGS-Handbook/">
            <meta name="twitter:title" content="NGS Data Analysis Handbook">
            <meta name="twitter:description" content="">
            <meta name="twitter:image" content="https://iliapopov17.github.io/images/ngs-handbook/metatag.png">

            <!-- Meta Tags Generated via https://www.opengraph.xyz -->
        </head>
        <body>
        </body>
    </html>

  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#iii-whole-genome-and-pangenome-analyses" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="NGS Analysis Handbook" class="md-header__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  <img src="../assets/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NGS Analysis Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              III. Whole Genome and Pangenome Analyses
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9zM20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zm-9.15 3.96h2.3L12 9z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/iliapopov17/NGS-Handbook" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.7.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    iliapopov17/NGS-Handbook
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="NGS Analysis Handbook" class="md-nav__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  <img src="../assets/logo.svg" alt="logo">

    </a>
    NGS Analysis Handbook
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/iliapopov17/NGS-Handbook" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.7.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    iliapopov17/NGS-Handbook
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../01_Quality_Control/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    I. Quality Control of Raw Data
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_Variant_Calling/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    II. Genomic Variation Analysis
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    III. Whole Genome and Pangenome Analyses
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    III. Whole Genome and Pangenome Analyses
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    <span class="md-ellipsis">
      Introduction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#task" class="md-nav__link">
    <span class="md-ellipsis">
      Task
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-download-genomes-from-the-database-and-check-their-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Download genomes from the database and check their quality
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-annotate-the-genomes" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Annotate the genomes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: Annotate the genomes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-21-proteins-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.1. Proteins directory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-22-lstinfo-lst-file" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.2. LSTINFO-.lst file
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-23-genes-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.3. Genes directory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-24-replicons-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.4. Replicons directory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-25-lstinfo-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.5. LSTINFO directory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-26-tmp_files-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.6. tmp_files directory
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-build-orthological-series" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Build orthological series
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-extract-pangenome" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Extract pangenome
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-aligning-the-sequence-of-core-gene-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Aligning the sequence of core gene sequences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 5: Aligning the sequence of core gene sequences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-51-align-esma-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5.1. Align-EsMa directory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-52-phylo-esma-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5.2. Phylo-EsMa directory
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-build-a-phylogenomic-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6: Build a phylogenomic tree
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 6: Build a phylogenomic tree">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-61-tree-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6.1. Tree visualization
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IV. Phylogenetics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            IV. Phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_00_Overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Overview
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_01_Intro_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Drawing Trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_02_Mining_Data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Working with NCBI
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_03_MSA/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Multiple Sequence Alignment
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_04_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Preparing the alignment and building trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_05_Root_Date/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Rooting and comparing trees; Dating
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_06_Phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_07_Visualization_Pro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Visualization Pro
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_08_Tanglegram/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tanglegram
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    V. 16S amplicon analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            V. 16S amplicon analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../V.%2016S_amplicon_analysis/05_00_Overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Overview
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../V.%2016S_amplicon_analysis/05_01_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    16S analysis with DADA2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../V.%2016S_amplicon_analysis/05_02_Crohns_disease/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Practice 1: Crohn's disease
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../V.%2016S_amplicon_analysis/05_03_Parkinsons_disease/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Practice 2: Parkinson's disease
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    <span class="md-ellipsis">
      Introduction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#task" class="md-nav__link">
    <span class="md-ellipsis">
      Task
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-download-genomes-from-the-database-and-check-their-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Download genomes from the database and check their quality
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-annotate-the-genomes" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Annotate the genomes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: Annotate the genomes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-21-proteins-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.1. Proteins directory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-22-lstinfo-lst-file" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.2. LSTINFO-.lst file
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-23-genes-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.3. Genes directory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-24-replicons-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.4. Replicons directory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-25-lstinfo-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.5. LSTINFO directory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-26-tmp_files-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.6. tmp_files directory
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-build-orthological-series" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Build orthological series
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-extract-pangenome" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Extract pangenome
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-aligning-the-sequence-of-core-gene-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Aligning the sequence of core gene sequences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 5: Aligning the sequence of core gene sequences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-51-align-esma-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5.1. Align-EsMa directory
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-52-phylo-esma-directory" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5.2. Phylo-EsMa directory
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-build-a-phylogenomic-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6: Build a phylogenomic tree
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 6: Build a phylogenomic tree">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-61-tree-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6.1. Tree visualization
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="iii-whole-genome-and-pangenome-analyses"><strong>III. Whole Genome and Pangenome Analyses</strong></h1>
<blockquote>
<p>This chapter contains a manual on whole genome and pangenome analyses.</p>
</blockquote>
<p>We will use the <code>PanACoTA</code> tool to perform the pangenome analysis.</p>
<h2 id="instruction"><strong>Instruction</strong></h2>
<p><img alt="CLT" src="https://img.shields.io/badge/Language-Command Line Tools &amp; Python-steelblue" /><br>
<img alt="IDE" src="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" /></p>
<p>Every step of this pipeline can be done in <code>terminal</code> except <strong>Step 6.1.</strong>, which needs <code>Python</code>.<br>
It is recommended to use <code>Jupyter Notebook</code>. Just write <code>!</code> in the beggining of each cell to make it understand <code>bash</code> commands.</p>
<p>To recreate any of the steps of this manual please install:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>wget<span class="w"> </span>https://github.com/iliapopov17/NGS-Handbook/raw/refs/heads/main/envs/panacota.yaml
</span></code></pre></div>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>panacota.yaml
</span></code></pre></div>
<p>And of cource do not forget to activate the envinronment!</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>conda<span class="w"> </span>activate<span class="w"> </span>panacota
</span></code></pre></div>
<hr />
<h2 id="introduction"><strong>Introduction</strong></h2>
<p>Here we will work with <code>PanACoTA</code> pipeline. This tool has many dependecies:</p>
<ul>
<li>For <code>prepare</code> module: <code>mash</code> (to filter genomes)</li>
<li>For <code>annotate</code> module: <code>prokka</code> and/or <code>prodigal</code> (to uniformly annotate your genomes)</li>
<li>For <code>pangenome</code> module: <code>mmseqs</code> (to generate pangenomes)</li>
<li>For <code>align</code> module: <code>mafft</code> (to align persistent genome)</li>
<li>For <code>tree</code> module: At least one of those softwares, to infer a phylogenetic tree:<ul>
<li><code>IQ Tree</code></li>
<li><code>FastTreeMP</code></li>
<li><code>FastME</code></li>
<li><code>Quicktree</code></li>
</ul>
</li>
</ul>
<p>All of them but one will be installed with the <code>panacota.yaml</code> conda environment.<br>
The last one - <code>mash</code> please install yourself.<br>
For more details please visit <a href="https://github.com/gem-pasteur/PanACoTA"><code>PanACoTA</code> GitHub repository</a></p>
<h2 id="task"><strong>Task</strong></h2>
<details class="info" open="open">
<summary>Info</summary>
<p>We've sequenced and assembled the genome.<br>
It's the genome of some organism, and we even know what kind of organism it is.<br></p>
<ul>
<li>We download the genomes of its close relatives from public databases.<br></li>
<li>We annotate them all together (so that there is a homogeneous annotation).<br></li>
<li>Build orthologous series.<br></li>
<li>Extract the pangenome.<br></li>
<li>Make genes alignment.<br></li>
<li>Build a phylogenetic tree.<br></li>
</ul>
<p>We'll be working with the <em>Shigella flexneri</em> genome.<br>
It's genetically nested within <em>Escherichia</em>, but it's a very aggressive pathogen.<br>
The same pathogenicity factors are found in <em>E. marmotae</em> isolated from other mammals.<br>
So our target is <em>Shigella flexneri</em>. And for comparison, we want to download the genomes of <em>E. marmotae</em> isolated from marmosets and analyse them further.</p>
</details>
<hr />
<h2 id="step-1-download-genomes-from-the-database-and-check-their-quality"><strong>Step 1: Download genomes from the database and check their quality</strong></h2>
<p>As it was mention in the <strong>Introduction</strong> we will run <code>PanACoTA</code> pipeline.
It has many dependecies and in the <code>panacota.yaml</code> cona environment all but one are already installed.
The one missing is <code>mash</code>.
It is needed for this step, but it is not available in <code>conda</code> or <code>pip</code>.
It is distributed as binary. To install it on <code>Ubuntu</code> run this command in your terminal:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>sudo apt install mash
</span></code></pre></div>
<p>For more details read the <a href="https://mash.readthedocs.io/en/latest/"><code>mash</code> tutorial</a></p>
<p><code>Mash</code> measures the distances between genomes:<br></p>
<ul>
<li>if the distances are any small (i.e. we have some copies).<br></li>
<li>if the distances are too large (i.e. genomes so far away that nothing can be counted on them further).<br></li>
</ul>
<p>The programme discards them (this is a configurable parameter).<br></p>
<details class="info" open="open">
<summary>Info</summary>
<ul>
<li><code>GenBank</code> - in general, all possible <code>.fasta</code> files uploaded to <code>NCBI's GenBank</code> database.<br></li>
<li><code>RefSeq</code> - an attempt to structure <code>GenBank</code>, a so-called ‘quality mark’ meaning that <code>NCBI</code> staff have analysed the data according to their own pipeline and approved its quality at a high level.<br></li>
</ul>
</details>
<p>Let's break down the command below in parts:<br></p>
<ul>
<li><code>PanACoTA</code> - the name of the tool we are running.<br></li>
<li><code>prepare</code> - one of the modules of this tool.<br></li>
<li>By key <code>-g</code> we pass what genomes we are downloading - quotes are very important.<br></li>
<li>By key <code>-s</code> we choose between GenBank and RefSeq.<br></li>
<li>By key <code>-l</code> we choose the level of assembly.<br></li>
</ul>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>PanACoTA<span class="w"> </span>prepare<span class="w"> </span>-g<span class="w"> </span><span class="s1">&#39;Escherichia marmotae&#39;</span><span class="w"> </span>-s<span class="w"> </span>refseq<span class="w"> </span>-l<span class="w"> </span><span class="nb">complete</span>
</span></code></pre></div>
<p>Let's see what we have after executing this command</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>ls<span class="w"> </span>Escherichia_marmotae
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>Database_init
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>LSTINFO-Escherichia_marmotae-filtered-0.0001_0.06.txt
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>PanACoTA_prepare_Escherichia_marmotae.log
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>PanACoTA_prepare_Escherichia_marmotae.log.details
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>PanACoTA_prepare_Escherichia_marmotae.log.err
</span><span id="__span-6-6"><a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a>assembly_summary-Escherichia_marmotae.txt
</span><span id="__span-6-7"><a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a>discarded-by-L90_nbcont-Escherichia_marmotae.lst
</span><span id="__span-6-8"><a id="__codelineno-6-8" name="__codelineno-6-8" href="#__codelineno-6-8"></a>discarded-by-minhash-Escherichia_marmotae-0.0001_0.06.txt
</span><span id="__span-6-9"><a id="__codelineno-6-9" name="__codelineno-6-9" href="#__codelineno-6-9"></a>mash_files
</span><span id="__span-6-10"><a id="__codelineno-6-10" name="__codelineno-6-10" href="#__codelineno-6-10"></a>refseq
</span><span id="__span-6-11"><a id="__codelineno-6-11" name="__codelineno-6-11" href="#__codelineno-6-11"></a>tmp_files
</span></code></pre></div>
<ul>
<li><code>PanACoTA_prepare_Escherichia_marmotae.log</code> - file with log of command execution.<br></li>
<li><code>PanACoTA_prepare_Escherichia_marmotae.log.details</code> - file with detailed log of command execution.<br></li>
<li><code>PanACoTA_prepare_Escherichia_marmotae.log.err</code> - error log (if there were any), it is always created, but if there were no errors, the file will be empty.<br></li>
<li><code>assembly_summary-Escherichia_marmotae.txt</code> - statistics on a specific assembly of <em>Escherichia marmotae</em>.<br></li>
<li><code>discarded-by-minhash-Escherichia_marmotae-0.0001_0.06.txt</code> - file with what <code>mash</code> discarded (it discarded anything closer than <code>0.0001</code> and anything further than <code>0.06</code>).<br></li>
<li><code>LSTINFO-Escherichia_marmotae-filtered-0.0001e_0.06.txt</code> - a list of all the files that <code>PanACoTA</code> fragmented.<br></li>
</ul>
<p>Now let's look in the <code>Database_init</code> directory.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>ls<span class="w"> </span>Escherichia_marmotae/Database_init
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<p><div class="language-text highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>GCF_002900365.1_ASM290036v1_genomic.fna
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>GCF_013636045.1_ASM1363604v1_genomic.fna
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>GCF_013636235.1_ASM1363623v1_genomic.fna
</span><span id="__span-8-4"><a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>GCF_013732895.1_ASM1373289v1_genomic.fna
</span><span id="__span-8-5"><a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a>GCF_013745515.1_ASM1374551v1_genomic.fna
</span><span id="__span-8-6"><a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a>GCF_013746655.1_ASM1374665v1_genomic.fna
</span><span id="__span-8-7"><a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a>GCF_022592155.1_ASM2259215v1_genomic.fna
</span><span id="__span-8-8"><a id="__codelineno-8-8" name="__codelineno-8-8" href="#__codelineno-8-8"></a>GCF_029717905.1_ASM2971790v1_genomic.fna
</span><span id="__span-8-9"><a id="__codelineno-8-9" name="__codelineno-8-9" href="#__codelineno-8-9"></a>GCF_029719265.1_ASM2971926v1_genomic.fna
</span><span id="__span-8-10"><a id="__codelineno-8-10" name="__codelineno-8-10" href="#__codelineno-8-10"></a>GCF_029962465.1_ASM2996246v1_genomic.fna
</span><span id="__span-8-11"><a id="__codelineno-8-11" name="__codelineno-8-11" href="#__codelineno-8-11"></a>GCF_037055335.1_ASM3705533v1_genomic.fna
</span><span id="__span-8-12"><a id="__codelineno-8-12" name="__codelineno-8-12" href="#__codelineno-8-12"></a>GCF_900636405.1_41767_E01_genomic.fna
</span><span id="__span-8-13"><a id="__codelineno-8-13" name="__codelineno-8-13" href="#__codelineno-8-13"></a>GCF_900637015.1_46514_C01_genomic.fna
</span></code></pre></div>
Thus <code>PanACoTA</code> downloaded 12 complete genomes (nucleotide genomic sequences) of <em>Escherichia marmotae</em> from <code>RefSeq</code> into the <code>Escherichia_marmotae/Database_init directory</code>.<br></p>
<p>Now we will add to all this our <em>Shigella flexneri</em>.</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>wget<span class="w"> </span>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/007/405/GCF_000007405.1_ASM740v1/GCF_000007405.1_ASM740v1_genomic.fna.gz<span class="w"> </span>-P<span class="w"> </span>Escherichia_marmotae/Database_init/
</span></code></pre></div>
<p>Let's unpack the archive.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>gzip<span class="w"> </span>-d<span class="w"> </span>Escherichia_marmotae/Database_init/GCF_000007405.1_ASM740v1_genomic.fna.gz
</span></code></pre></div>
<p>The next step of <code>PanACoTA</code> takes as input a text file with a list of files containing genomes.<br>
That's why we'll create it!</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>ls<span class="w"> </span>Escherichia_marmotae/Database_init/<span class="w"> </span>&gt;<span class="w"> </span>listFile<span class="w"> </span>
</span></code></pre></div>
<p>Let's check this list</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>cat<span class="w"> </span>listFile
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>GCF_000007405.1_ASM740v1_genomic.fna
</span><span id="__span-13-2"><a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>GCF_002900365.1_ASM290036v1_genomic.fna
</span><span id="__span-13-3"><a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a>GCF_013636045.1_ASM1363604v1_genomic.fna
</span><span id="__span-13-4"><a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a>GCF_013636235.1_ASM1363623v1_genomic.fna
</span><span id="__span-13-5"><a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a>GCF_013732895.1_ASM1373289v1_genomic.fna
</span><span id="__span-13-6"><a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a>GCF_013745515.1_ASM1374551v1_genomic.fna
</span><span id="__span-13-7"><a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a>GCF_013746655.1_ASM1374665v1_genomic.fna
</span><span id="__span-13-8"><a id="__codelineno-13-8" name="__codelineno-13-8" href="#__codelineno-13-8"></a>GCF_022592155.1_ASM2259215v1_genomic.fna
</span><span id="__span-13-9"><a id="__codelineno-13-9" name="__codelineno-13-9" href="#__codelineno-13-9"></a>GCF_029717905.1_ASM2971790v1_genomic.fna
</span><span id="__span-13-10"><a id="__codelineno-13-10" name="__codelineno-13-10" href="#__codelineno-13-10"></a>GCF_029719265.1_ASM2971926v1_genomic.fna
</span><span id="__span-13-11"><a id="__codelineno-13-11" name="__codelineno-13-11" href="#__codelineno-13-11"></a>GCF_029962465.1_ASM2996246v1_genomic.fna
</span><span id="__span-13-12"><a id="__codelineno-13-12" name="__codelineno-13-12" href="#__codelineno-13-12"></a>GCF_037055335.1_ASM3705533v1_genomic.fna
</span><span id="__span-13-13"><a id="__codelineno-13-13" name="__codelineno-13-13" href="#__codelineno-13-13"></a>GCF_900636405.1_41767_E01_genomic.fna
</span><span id="__span-13-14"><a id="__codelineno-13-14" name="__codelineno-13-14" href="#__codelineno-13-14"></a>GCF_900637015.1_46514_C01_genomic.fna
</span></code></pre></div>
<p>That's right!<br>
These are the files that will be given as input in the next step of <code>PanACoTA</code> pipeline.<br></p>
<hr />
<h2 id="step-2-annotate-the-genomes"><strong>Step 2: Annotate the genomes</strong></h2>
<p>Running <code>PanACoTA</code> with the <code>annotate</code> mode.<br>
This mode of <code>PanACoTA</code> is basically a shell for using the <code>prokka</code> tool.<br>
Let's also break down the command below in parts:<br></p>
<ul>
<li><code>PanACoTA</code> - the name of the tool we are running.<br></li>
<li><code>annotate</code> - module of this tool that annotates genomes.<br></li>
<li>By key <code>-d</code> we pass the directory with the genomes.<br></li>
<li>By key <code>-r</code> we specify where to store the output files.<br></li>
<li>By key <code>-n</code> we set some four-character name for the data, which will then be used in the pipeline.<br></li>
<li>By key <code>-l</code> we specify what exactly genomes from the directory in <code>-d</code> key must be annotated.<br></li>
<li>By key <code>--threads</code> we specify how much threads this command can use (if use <code>0</code> - automatically uses all available cores).<br></li>
</ul>
<details class="tip" open="open">
<summary>Tip</summary>
<p>It takes a long time to execute the command (it took me 19 minutes)</p>
</details>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a>PanACoTA<span class="w"> </span>annotate<span class="w"> </span>-d<span class="w"> </span>Escherichia_marmotae/Database_init/<span class="w"> </span>-r<span class="w"> </span>Annotation<span class="w"> </span>-n<span class="w"> </span>EsMa<span class="w"> </span>-l<span class="w"> </span>listFile<span class="w"> </span>--threads<span class="w"> </span><span class="m">24</span>
</span></code></pre></div>
<p>Let's see what we got as a result of the annotation</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a>ls<span class="w"> </span>Annotation
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a>Genes                   QC_L90-listFile.png
</span><span id="__span-16-2"><a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a>LSTINFO                 QC_nb-contigs-listFile.png
</span><span id="__span-16-3"><a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a>LSTINFO-.lst                Replicons
</span><span id="__span-16-4"><a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a>PanACoTA-annotate_listFile.log      discarded-.lst
</span><span id="__span-16-5"><a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a>PanACoTA-annotate_listFile.log.details  gff3
</span><span id="__span-16-6"><a id="__codelineno-16-6" name="__codelineno-16-6" href="#__codelineno-16-6"></a>PanACoTA-annotate_listFile.log.err  tmp_files
</span><span id="__span-16-7"><a id="__codelineno-16-7" name="__codelineno-16-7" href="#__codelineno-16-7"></a>Proteins
</span></code></pre></div>
<p>We now have an <code>Annotation</code> folder, and in it:<br></p>
<ul>
<li><code>Proteins</code><br></li>
<li><code>Genes</code><br></li>
<li><code>Replicons</code><br></li>
<li><code>gff3</code><br></li>
<li><code>LSTINFO</code><br></li>
</ul>
<h3 id="step-21-proteins-directory"><strong>Step 2.1. <code>Proteins</code> directory</strong></h3>
<p>Let's look at the <code>Annotation/Proteins/</code> directory. What is stored in it?</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a>ls<span class="w"> </span>Annotation/Proteins/
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-18-1"><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a>EsMa.0824.00001.prt  EsMa.0824.00006.prt  EsMa.0824.00011.prt
</span><span id="__span-18-2"><a id="__codelineno-18-2" name="__codelineno-18-2" href="#__codelineno-18-2"></a>EsMa.0824.00002.prt  EsMa.0824.00007.prt  EsMa.0824.00012.prt
</span><span id="__span-18-3"><a id="__codelineno-18-3" name="__codelineno-18-3" href="#__codelineno-18-3"></a>EsMa.0824.00003.prt  EsMa.0824.00008.prt  EsMa.0824.00013.prt
</span><span id="__span-18-4"><a id="__codelineno-18-4" name="__codelineno-18-4" href="#__codelineno-18-4"></a>EsMa.0824.00004.prt  EsMa.0824.00009.prt  EsMa.0824.00014.prt
</span><span id="__span-18-5"><a id="__codelineno-18-5" name="__codelineno-18-5" href="#__codelineno-18-5"></a>EsMa.0824.00005.prt  EsMa.0824.00010.prt  EsMa.All.prt
</span></code></pre></div>
<p>So, we have 15 files in this folder (14 files for each organism and 1 common file).<br></p>
<ul>
<li><code>EsMa</code> is the name we set in the <code>-n</code> key.<br></li>
<li><code>0824</code> is the launch ID - August 24.<br></li>
<li><code>00001-00014</code> - organism sequence number (IMPORTANT: the files are not always in the exact order in which they were uploaded to listFile).<br></li>
</ul>
<p>Let's take a look at <code>Annotation/Proteins/EsMa.0824.00001.prt</code>.</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-19-1"><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a>head<span class="w"> </span>-10<span class="w"> </span>Annotation/Proteins/EsMa.0824.00001.prt
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-20-1"><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a>&gt;EsMa.0824.00001.0001b_00001 2463 thrA | Bifunctional aspartokinase/homoserine dehydrogenase 1 | NA | similar to AA sequence:UniProtKB:P00561 | COG:COG0460
</span><span id="__span-20-2"><a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a>MRVLKFGGTSVANAERFLRVADILESNARQGQVATVLSAPAKITNHLVAMIEKTISGQDA
</span><span id="__span-20-3"><a id="__codelineno-20-3" name="__codelineno-20-3" href="#__codelineno-20-3"></a>LPNISDAERIFAELLTGLAAAQPGFPLAQLKTFVDQEFAQIKHVLHGISLLGQCPDSINA
</span><span id="__span-20-4"><a id="__codelineno-20-4" name="__codelineno-20-4" href="#__codelineno-20-4"></a>ALICRGEKMSIAIMAGVLEARGHNVTVIDPVEKLLAVGHYLESTVDIAESTRRIAASRIP
</span><span id="__span-20-5"><a id="__codelineno-20-5" name="__codelineno-20-5" href="#__codelineno-20-5"></a>ADHMVLMAGFTAGNEKGELVVLGRNGSDYSAAVLAACLRADCCEIWTDVDGVYTCDPRQV
</span><span id="__span-20-6"><a id="__codelineno-20-6" name="__codelineno-20-6" href="#__codelineno-20-6"></a>PDARLLKSMSYQEAMELSYFGAKVLHPRTITPIAQFQIPCLIKNTGNPQAPGTLIGASRD
</span><span id="__span-20-7"><a id="__codelineno-20-7" name="__codelineno-20-7" href="#__codelineno-20-7"></a>EDELPVKGISNLNNMAMFSVSGPGMKGMVGMAARVFAAMSRARISVVLITQSSSEYSISF
</span><span id="__span-20-8"><a id="__codelineno-20-8" name="__codelineno-20-8" href="#__codelineno-20-8"></a>CVPQSDCVRAERAMQEEFYLELKEGLLEPLAVTERLAIISVVGDGMRTLRGISAKFFAAL
</span><span id="__span-20-9"><a id="__codelineno-20-9" name="__codelineno-20-9" href="#__codelineno-20-9"></a>ARANINIVAIAQGSSERSISVVVNNDDATTGVRVTHQMLFNTDQVIEVFVIGVGGVGGAL
</span><span id="__span-20-10"><a id="__codelineno-20-10" name="__codelineno-20-10" href="#__codelineno-20-10"></a>LEQLKRQQSWLKNKHIDLRVCGVANSKALLTSVHGLNLENWQEELAQAKEPFNLGRLIRL
</span></code></pre></div>
<p>And inside each <code>.prt</code> file, we see the amino acid sequences of all the proteins that <code>PanACoTA</code> predicted for us!<br>
Let's turn our attention to line 1:<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-21-1"><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a>head<span class="w"> </span>-1<span class="w"> </span>Annotation/Proteins/EsMa.0824.00001.prt
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-22-1"><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a>&gt;EsMa.0824.00001.0001b_00001 2463 thrA | Bifunctional aspartokinase/homoserine dehydrogenase 1 | NA | similar to AA sequence:UniProtKB:P00561 | COG:COG0460
</span></code></pre></div>
<ul>
<li><code>&gt;EsMa.0824.00001.</code> - the name of the genome in the <code>PanACoTA</code>'s annotation.<br></li>
<li><code>0001b</code> - replicon number.<br></li>
<li><code>_00001</code> - frame number.<br></li>
<li><code>2463</code> - length.<br></li>
<li><code>thrA</code> - protein's name.<br></li>
<li><code>Bifunctional aspartokinase/homoserine dehydrogenase 1</code> - protein's function.<br></li>
<li><code>similar to AA sequence:UniProtKB:P00561</code> - the basis for predicting the protein and its function.<br></li>
<li><code>COG:COG0460</code> - COG ID.<br></li>
</ul>
<h3 id="step-22-lstinfo-lst-file"><strong>Step 2.2. <code>LSTINFO-.lst</code> file</strong></h3>
<p>Now, let's look at the <code>Annotation/LSTINFO-.lst</code> file.</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-23-1"><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a>cat<span class="w"> </span>Annotation/LSTINFO-.lst
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-24-1"><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a>gembase_name    orig_name   to_annotate gsize   nb_conts    L90
</span><span id="__span-24-2"><a id="__codelineno-24-2" name="__codelineno-24-2" href="#__codelineno-24-2"></a>EsMa.0824.00001 GCF_000007405.1_ASM740v1_genomic.fna    Annotation/tmp_files/GCF_000007405.1_ASM740v1_genomic.fna_prokka-split5N.fna    4599354 1   1
</span><span id="__span-24-3"><a id="__codelineno-24-3" name="__codelineno-24-3" href="#__codelineno-24-3"></a>EsMa.0824.00002 GCF_900636405.1_41767_E01_genomic.fna   Annotation/tmp_files/GCF_900636405.1_41767_E01_genomic.fna_prokka-split5N.fna   4857140 1   1
</span><span id="__span-24-4"><a id="__codelineno-24-4" name="__codelineno-24-4" href="#__codelineno-24-4"></a>EsMa.0824.00003 GCF_900637015.1_46514_C01_genomic.fna   Annotation/tmp_files/GCF_900637015.1_46514_C01_genomic.fna_prokka-split5N.fna   4450344 1   1
</span><span id="__span-24-5"><a id="__codelineno-24-5" name="__codelineno-24-5" href="#__codelineno-24-5"></a>EsMa.0824.00004 GCF_013636045.1_ASM1363604v1_genomic.fna    Annotation/tmp_files/GCF_013636045.1_ASM1363604v1_genomic.fna_prokka-split5N.fna    4637518 2   1
</span><span id="__span-24-6"><a id="__codelineno-24-6" name="__codelineno-24-6" href="#__codelineno-24-6"></a>EsMa.0824.00005 GCF_029962465.1_ASM2996246v1_genomic.fna    Annotation/tmp_files/GCF_029962465.1_ASM2996246v1_genomic.fna_prokka-split5N.fna    4669757 2   1
</span><span id="__span-24-7"><a id="__codelineno-24-7" name="__codelineno-24-7" href="#__codelineno-24-7"></a>EsMa.0824.00006 GCF_002900365.1_ASM290036v1_genomic.fna Annotation/tmp_files/GCF_002900365.1_ASM290036v1_genomic.fna_prokka-split5N.fna 4896291 3   1
</span><span id="__span-24-8"><a id="__codelineno-24-8" name="__codelineno-24-8" href="#__codelineno-24-8"></a>EsMa.0824.00007 GCF_013636235.1_ASM1363623v1_genomic.fna    Annotation/tmp_files/GCF_013636235.1_ASM1363623v1_genomic.fna_prokka-split5N.fna    4981845 3   1
</span><span id="__span-24-9"><a id="__codelineno-24-9" name="__codelineno-24-9" href="#__codelineno-24-9"></a>EsMa.0824.00008 GCF_013745515.1_ASM1374551v1_genomic.fna    Annotation/tmp_files/GCF_013745515.1_ASM1374551v1_genomic.fna_prokka-split5N.fna    4887958 3   1
</span><span id="__span-24-10"><a id="__codelineno-24-10" name="__codelineno-24-10" href="#__codelineno-24-10"></a>EsMa.0824.00009 GCF_022592155.1_ASM2259215v1_genomic.fna    Annotation/tmp_files/GCF_022592155.1_ASM2259215v1_genomic.fna_prokka-split5N.fna    5200167 3   1
</span><span id="__span-24-11"><a id="__codelineno-24-11" name="__codelineno-24-11" href="#__codelineno-24-11"></a>EsMa.0824.00010 GCF_013746655.1_ASM1374665v1_genomic.fna    Annotation/tmp_files/GCF_013746655.1_ASM1374665v1_genomic.fna_prokka-split5N.fna    5094503 4   1
</span><span id="__span-24-12"><a id="__codelineno-24-12" name="__codelineno-24-12" href="#__codelineno-24-12"></a>EsMa.0824.00011 GCF_037055335.1_ASM3705533v1_genomic.fna    Annotation/tmp_files/GCF_037055335.1_ASM3705533v1_genomic.fna_prokka-split5N.fna    5022908 4   1
</span><span id="__span-24-13"><a id="__codelineno-24-13" name="__codelineno-24-13" href="#__codelineno-24-13"></a>EsMa.0824.00012 GCF_029719265.1_ASM2971926v1_genomic.fna    Annotation/tmp_files/GCF_029719265.1_ASM2971926v1_genomic.fna_prokka-split5N.fna    5075061 5   1
</span><span id="__span-24-14"><a id="__codelineno-24-14" name="__codelineno-24-14" href="#__codelineno-24-14"></a>EsMa.0824.00013 GCF_013732895.1_ASM1373289v1_genomic.fna    Annotation/tmp_files/GCF_013732895.1_ASM1373289v1_genomic.fna_prokka-split5N.fna    5083937 6   1
</span><span id="__span-24-15"><a id="__codelineno-24-15" name="__codelineno-24-15" href="#__codelineno-24-15"></a>EsMa.0824.00014 GCF_029717905.1_ASM2971790v1_genomic.fna    Annotation/tmp_files/GCF_029717905.1_ASM2971790v1_genomic.fna_prokka-split5N.fna    4780041 7   1
</span></code></pre></div>
<ul>
<li><code>EsMa.0824.00001</code> - the name that <code>PanACoTA</code> renamed the file to (by <code>-n</code> key).<br></li>
<li><code>GCF_000007405.1_ASM740v1_genomic.fna</code> - the name from which <code>PanACoTA</code> was renaming.<br></li>
<li><code>Annotation/tmp_files/GCF_000007405.1_ASM740v1_genomic.fna_prokka-split5N.fna</code>- a folder containing what <code>PanACoTA</code> eventually annotated.<br></li>
<li><code>4599354</code> - genome size.<br></li>
</ul>
<h3 id="step-23-genes-directory"><strong>Step 2.3. <code>Genes</code> directory</strong></h3>
<p>Let's take a look at the contents of the <code>Genes</code> directory.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-25-1"><a id="__codelineno-25-1" name="__codelineno-25-1" href="#__codelineno-25-1"></a>ls<span class="w"> </span>Annotation/Genes/
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-26-1"><a id="__codelineno-26-1" name="__codelineno-26-1" href="#__codelineno-26-1"></a>EsMa.0824.00001.gen  EsMa.0824.00006.gen  EsMa.0824.00011.gen
</span><span id="__span-26-2"><a id="__codelineno-26-2" name="__codelineno-26-2" href="#__codelineno-26-2"></a>EsMa.0824.00002.gen  EsMa.0824.00007.gen  EsMa.0824.00012.gen
</span><span id="__span-26-3"><a id="__codelineno-26-3" name="__codelineno-26-3" href="#__codelineno-26-3"></a>EsMa.0824.00003.gen  EsMa.0824.00008.gen  EsMa.0824.00013.gen
</span><span id="__span-26-4"><a id="__codelineno-26-4" name="__codelineno-26-4" href="#__codelineno-26-4"></a>EsMa.0824.00004.gen  EsMa.0824.00009.gen  EsMa.0824.00014.gen
</span><span id="__span-26-5"><a id="__codelineno-26-5" name="__codelineno-26-5" href="#__codelineno-26-5"></a>EsMa.0824.00005.gen  EsMa.0824.00010.gen
</span></code></pre></div>
<p>So we have 14 files in this folder (14 files for each organism).<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-27-1"><a id="__codelineno-27-1" name="__codelineno-27-1" href="#__codelineno-27-1"></a>head<span class="w"> </span>-10<span class="w"> </span>Annotation/Genes/EsMa.0824.00001.gen
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-28-1"><a id="__codelineno-28-1" name="__codelineno-28-1" href="#__codelineno-28-1"></a>&gt;EsMa.0824.00001.0001b_00001 2463 thrA | Bifunctional aspartokinase/homoserine dehydrogenase 1 | NA | similar to AA sequence:UniProtKB:P00561 | COG:COG0460
</span><span id="__span-28-2"><a id="__codelineno-28-2" name="__codelineno-28-2" href="#__codelineno-28-2"></a>ATGCGAGTGTTGAAGTTCGGCGGTACATCAGTGGCAAATGCAGAACGTTTTCTGCGTGTT
</span><span id="__span-28-3"><a id="__codelineno-28-3" name="__codelineno-28-3" href="#__codelineno-28-3"></a>GCCGATATTCTGGAAAGCAATGCCAGGCAGGGGCAGGTGGCCACCGTCCTCTCTGCCCCC
</span><span id="__span-28-4"><a id="__codelineno-28-4" name="__codelineno-28-4" href="#__codelineno-28-4"></a>GCCAAAATCACCAACCACCTGGTGGCGATGATTGAAAAAACCATTAGCGGCCAGGATGCT
</span><span id="__span-28-5"><a id="__codelineno-28-5" name="__codelineno-28-5" href="#__codelineno-28-5"></a>TTACCCAATATCAGCGATGCCGAACGTATTTTTGCCGAACTTTTGACGGGACTCGCCGCC
</span><span id="__span-28-6"><a id="__codelineno-28-6" name="__codelineno-28-6" href="#__codelineno-28-6"></a>GCCCAGCCGGGGTTCCCGCTGGCGCAATTGAAAACTTTCGTCGATCAGGAATTTGCCCAA
</span><span id="__span-28-7"><a id="__codelineno-28-7" name="__codelineno-28-7" href="#__codelineno-28-7"></a>ATAAAACATGTCCTGCATGGCATTAGTTTGTTGGGGCAGTGCCCGGATAGCATCAACGCT
</span><span id="__span-28-8"><a id="__codelineno-28-8" name="__codelineno-28-8" href="#__codelineno-28-8"></a>GCGCTGATTTGCCGTGGCGAGAAAATGTCGATCGCCATTATGGCCGGCGTGTTAGAAGCG
</span><span id="__span-28-9"><a id="__codelineno-28-9" name="__codelineno-28-9" href="#__codelineno-28-9"></a>CGTGGTCACAACGTTACCGTTATCGATCCGGTCGAAAAACTGCTGGCAGTGGGGCATTAC
</span><span id="__span-28-10"><a id="__codelineno-28-10" name="__codelineno-28-10" href="#__codelineno-28-10"></a>CTCGAATCTACCGTCGATATTGCTGAGTCCACCCGCCGTATTGCGGCAAGCCGCATTCCG
</span></code></pre></div>
<p>So here we have nucleotide sequences! The names of genes and proteins are the same!<br></p>
<h3 id="step-24-replicons-directory"><strong>Step 2.4. <code>Replicons</code> directory</strong></h3>
<p>We also have a directory called <code>Replicons</code>, but it's a boring directory that just holds actually the original nucleotide sequences, but just under code numbers specified in the <code>-n</code> key.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-29-1"><a id="__codelineno-29-1" name="__codelineno-29-1" href="#__codelineno-29-1"></a>ls<span class="w"> </span>Annotation/Replicons/
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-30-1"><a id="__codelineno-30-1" name="__codelineno-30-1" href="#__codelineno-30-1"></a>EsMa.0824.00001.fna  EsMa.0824.00006.fna  EsMa.0824.00011.fna
</span><span id="__span-30-2"><a id="__codelineno-30-2" name="__codelineno-30-2" href="#__codelineno-30-2"></a>EsMa.0824.00002.fna  EsMa.0824.00007.fna  EsMa.0824.00012.fna
</span><span id="__span-30-3"><a id="__codelineno-30-3" name="__codelineno-30-3" href="#__codelineno-30-3"></a>EsMa.0824.00003.fna  EsMa.0824.00008.fna  EsMa.0824.00013.fna
</span><span id="__span-30-4"><a id="__codelineno-30-4" name="__codelineno-30-4" href="#__codelineno-30-4"></a>EsMa.0824.00004.fna  EsMa.0824.00009.fna  EsMa.0824.00014.fna
</span><span id="__span-30-5"><a id="__codelineno-30-5" name="__codelineno-30-5" href="#__codelineno-30-5"></a>EsMa.0824.00005.fna  EsMa.0824.00010.fna
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-31-1"><a id="__codelineno-31-1" name="__codelineno-31-1" href="#__codelineno-31-1"></a>head<span class="w"> </span>Annotation/Replicons/EsMa.0824.00001.fna
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-32-1"><a id="__codelineno-32-1" name="__codelineno-32-1" href="#__codelineno-32-1"></a>&gt;EsMa.0824.00001.0001 4599354
</span><span id="__span-32-2"><a id="__codelineno-32-2" name="__codelineno-32-2" href="#__codelineno-32-2"></a>AGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGAGTGTC
</span><span id="__span-32-3"><a id="__codelineno-32-3" name="__codelineno-32-3" href="#__codelineno-32-3"></a>TGATAGCAGCTTCTGAACTGGTTACCTGCCGTGAGTAAATTAAAATTTTATTGACTTAGG
</span><span id="__span-32-4"><a id="__codelineno-32-4" name="__codelineno-32-4" href="#__codelineno-32-4"></a>TCACTAAATACTTTAACCAATATAGGCATAGCGCACAGACAGATAAAAATTACAGAGTAC
</span><span id="__span-32-5"><a id="__codelineno-32-5" name="__codelineno-32-5" href="#__codelineno-32-5"></a>ACAACATCCATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGT
</span><span id="__span-32-6"><a id="__codelineno-32-6" name="__codelineno-32-6" href="#__codelineno-32-6"></a>AACGGTGCGGGCTGACGCGTACAGGAAACACAGAAAAAAGCCCGCACCTGACAGTGCGGG
</span><span id="__span-32-7"><a id="__codelineno-32-7" name="__codelineno-32-7" href="#__codelineno-32-7"></a>CTTTTTTTTCGACCAAAGGTAACGAGGTAACAACCATGCGAGTGTTGAAGTTCGGCGGTA
</span><span id="__span-32-8"><a id="__codelineno-32-8" name="__codelineno-32-8" href="#__codelineno-32-8"></a>CATCAGTGGCAAATGCAGAACGTTTTCTGCGTGTTGCCGATATTCTGGAAAGCAATGCCA
</span><span id="__span-32-9"><a id="__codelineno-32-9" name="__codelineno-32-9" href="#__codelineno-32-9"></a>GGCAGGGGCAGGTGGCCACCGTCCTCTCTGCCCCCGCCAAAATCACCAACCACCTGGTGG
</span><span id="__span-32-10"><a id="__codelineno-32-10" name="__codelineno-32-10" href="#__codelineno-32-10"></a>CGATGATTGAAAAAACCATTAGCGGCCAGGATGCTTTACCCAATATCAGCGATGCCGAAC
</span></code></pre></div>
<h3 id="step-25-lstinfo-directory"><strong>Step 2.5. <code>LSTINFO</code> directory</strong></h3>
<p>And we have a directory called <code>LSTINFO</code>.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-33-1"><a id="__codelineno-33-1" name="__codelineno-33-1" href="#__codelineno-33-1"></a>ls<span class="w"> </span>Annotation/LSTINFO/
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-34-1"><a id="__codelineno-34-1" name="__codelineno-34-1" href="#__codelineno-34-1"></a>EsMa.0824.00001.lst  EsMa.0824.00006.lst  EsMa.0824.00011.lst
</span><span id="__span-34-2"><a id="__codelineno-34-2" name="__codelineno-34-2" href="#__codelineno-34-2"></a>EsMa.0824.00002.lst  EsMa.0824.00007.lst  EsMa.0824.00012.lst
</span><span id="__span-34-3"><a id="__codelineno-34-3" name="__codelineno-34-3" href="#__codelineno-34-3"></a>EsMa.0824.00003.lst  EsMa.0824.00008.lst  EsMa.0824.00013.lst
</span><span id="__span-34-4"><a id="__codelineno-34-4" name="__codelineno-34-4" href="#__codelineno-34-4"></a>EsMa.0824.00004.lst  EsMa.0824.00009.lst  EsMa.0824.00014.lst
</span><span id="__span-34-5"><a id="__codelineno-34-5" name="__codelineno-34-5" href="#__codelineno-34-5"></a>EsMa.0824.00005.lst  EsMa.0824.00010.lst
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-35-1"><a id="__codelineno-35-1" name="__codelineno-35-1" href="#__codelineno-35-1"></a>head<span class="w"> </span>-10<span class="w"> </span>Annotation/LSTINFO/EsMa.0824.00001.lst
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-36-1"><a id="__codelineno-36-1" name="__codelineno-36-1" href="#__codelineno-36-1"></a>336 2798    D   CDS EsMa.0824.00001.0001b_00001 thrA    | Bifunctional aspartokinase/homoserine dehydrogenase 1 | NA | similar to AA sequence:UniProtKB:P00561 | COG:COG0460
</span><span id="__span-36-2"><a id="__codelineno-36-2" name="__codelineno-36-2" href="#__codelineno-36-2"></a>2800    3732    D   CDS EsMa.0824.00001.0001i_00002 thrB    | Homoserine kinase | 2.7.1.39 | similar to AA sequence:UniProtKB:P00547 | COG:COG0083
</span><span id="__span-36-3"><a id="__codelineno-36-3" name="__codelineno-36-3" href="#__codelineno-36-3"></a>3733    5019    D   CDS EsMa.0824.00001.0001i_00003 thrC    | Threonine synthase | 4.2.3.1 | similar to AA sequence:UniProtKB:P00934 | COG:COG0498
</span><span id="__span-36-4"><a id="__codelineno-36-4" name="__codelineno-36-4" href="#__codelineno-36-4"></a>5233    5529    D   CDS EsMa.0824.00001.0001i_00004 NA  | hypothetical protein | NA | NA | NA
</span><span id="__span-36-5"><a id="__codelineno-36-5" name="__codelineno-36-5" href="#__codelineno-36-5"></a>5682    6458    C   CDS EsMa.0824.00001.0001i_00005 yaaA    | Peroxide stress resistance protein YaaA | NA | similar to AA sequence:UniProtKB:P0A8I3 | COG:COG3022
</span><span id="__span-36-6"><a id="__codelineno-36-6" name="__codelineno-36-6" href="#__codelineno-36-6"></a>6528    6896    C   CDS EsMa.0824.00001.0001i_00006 alsT_1  | Amino-acid carrier protein AlsT | NA | similar to AA sequence:UniProtKB:Q45068 | COG:COG1115
</span><span id="__span-36-7"><a id="__codelineno-36-7" name="__codelineno-36-7" href="#__codelineno-36-7"></a>6918    7958    C   CDS EsMa.0824.00001.0001i_00007 alsT_2  | Amino-acid carrier protein AlsT | NA | similar to AA sequence:UniProtKB:Q45068 | COG:COG1115
</span><span id="__span-36-8"><a id="__codelineno-36-8" name="__codelineno-36-8" href="#__codelineno-36-8"></a>8237    9190    D   CDS EsMa.0824.00001.0001i_00008 talB    | Transaldolase B | 2.2.1.2 | similar to AA sequence:UniProtKB:P0A870 | COG:COG0176
</span><span id="__span-36-9"><a id="__codelineno-36-9" name="__codelineno-36-9" href="#__codelineno-36-9"></a>9305    9892    D   CDS EsMa.0824.00001.0001i_00009 mog | Molybdopterin adenylyltransferase | 2.7.7.75 | similar to AA sequence:UniProtKB:P0AF03 | COG:COG0521
</span><span id="__span-36-10"><a id="__codelineno-36-10" name="__codelineno-36-10" href="#__codelineno-36-10"></a>9927    10493   C   CDS EsMa.0824.00001.0001i_00010 satP    | Succinate-acetate/proton symporter SatP | NA | similar to AA sequence:UniProtKB:P0AC98 | COG:COG1584
</span></code></pre></div>
<p>This is a tabular format of all annotated features in the organism!<br>
This is especially convenient if we work with genomes using programming languages (e.g. <code>Python</code> and <code>pandas</code>).<br></p>
<h3 id="step-26-tmp_files-directory"><strong>Step 2.6. <code>tmp_files</code> directory</strong></h3>
<p><strong><em>Input</em></strong></p>
<p><div class="language-bash highlight"><pre><span></span><code><span id="__span-37-1"><a id="__codelineno-37-1" name="__codelineno-37-1" href="#__codelineno-37-1"></a>ls<span class="w"> </span>-U<span class="w"> </span>Annotation/tmp_files/<span class="w"> </span><span class="p">|</span><span class="w"> </span>head<span class="w"> </span>-10
</span></code></pre></div>
<strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-38-1"><a id="__codelineno-38-1" name="__codelineno-38-1" href="#__codelineno-38-1"></a>GCF_000007405.1_ASM740v1_genomic.fna_prokka-split5N.fna
</span><span id="__span-38-2"><a id="__codelineno-38-2" name="__codelineno-38-2" href="#__codelineno-38-2"></a>GCF_000007405.1_ASM740v1_genomic.fna_prokka-split5N.fna-prokka.log
</span><span id="__span-38-3"><a id="__codelineno-38-3" name="__codelineno-38-3" href="#__codelineno-38-3"></a>GCF_000007405.1_ASM740v1_genomic.fna_prokka-split5N.fna-prokkaRes
</span><span id="__span-38-4"><a id="__codelineno-38-4" name="__codelineno-38-4" href="#__codelineno-38-4"></a>GCF_002900365.1_ASM290036v1_genomic.fna_prokka-split5N.fna
</span><span id="__span-38-5"><a id="__codelineno-38-5" name="__codelineno-38-5" href="#__codelineno-38-5"></a>GCF_002900365.1_ASM290036v1_genomic.fna_prokka-split5N.fna-prokka.log
</span><span id="__span-38-6"><a id="__codelineno-38-6" name="__codelineno-38-6" href="#__codelineno-38-6"></a>GCF_002900365.1_ASM290036v1_genomic.fna_prokka-split5N.fna-prokkaRes
</span><span id="__span-38-7"><a id="__codelineno-38-7" name="__codelineno-38-7" href="#__codelineno-38-7"></a>GCF_013636045.1_ASM1363604v1_genomic.fna_prokka-split5N.fna
</span><span id="__span-38-8"><a id="__codelineno-38-8" name="__codelineno-38-8" href="#__codelineno-38-8"></a>GCF_013636045.1_ASM1363604v1_genomic.fna_prokka-split5N.fna-prokka.log
</span><span id="__span-38-9"><a id="__codelineno-38-9" name="__codelineno-38-9" href="#__codelineno-38-9"></a>GCF_013636045.1_ASM1363604v1_genomic.fna_prokka-split5N.fna-prokkaRes
</span><span id="__span-38-10"><a id="__codelineno-38-10" name="__codelineno-38-10" href="#__codelineno-38-10"></a>GCF_013636235.1_ASM1363623v1_genomic.fna_prokka-split5N.fna
</span></code></pre></div>
<p>The <code>tmp_files</code> directory contains logs on temporary files.<br>
What is it for?<br>
If, let's say, something that should be there is not found in the genome under study, the first thing to do is to study these files - most likely <code>prokka</code> crashed with an error and you can read the details in these temporary files.<br></p>
<hr />
<h2 id="step-3-build-orthological-series"><strong>Step 3: Build orthological series</strong></h2>
<p>This command builds the pangenome.<br>
Let's also break down the command below in parts:<br></p>
<ul>
<li><code>PanACoTA</code> - the name of the tool we are running.<br></li>
<li><code>pangenome</code> - module of this tool that builds the pangenome.<br></li>
<li>By key <code>-l</code> we pass the <code>LSTINFO-.lst</code> file from the <strong>Step 2.2.</strong>.<br></li>
<li>By key <code>-n</code> we pass the four-character name for the data, which was set previously.<br></li>
<li>By key <code>-d</code> we pass the directory with the proteins (since <code>PanACoTA</code> builds the pangenome by protein sequence similarity, it only needs the protein directory).<br></li>
<li>By key <code>-o</code> we specify where to store the output files.<br></li>
<li>By the key <code>-i</code> we establish the level of similarity (identity) of the sequences.<br></li>
</ul>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-39-1"><a id="__codelineno-39-1" name="__codelineno-39-1" href="#__codelineno-39-1"></a>PanACoTA<span class="w"> </span>pangenome<span class="w"> </span>-l<span class="w"> </span>Annotation/LSTINFO-.lst<span class="w"> </span>-n<span class="w"> </span>EsMa<span class="w"> </span>-d<span class="w"> </span>Annotation/Proteins/<span class="w"> </span>-o<span class="w"> </span>Pangenome<span class="w"> </span>-i<span class="w"> </span><span class="m">0</span>.8<span class="w"> </span>
</span></code></pre></div>
<p>Let's take a look at the log file.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-40-1"><a id="__codelineno-40-1" name="__codelineno-40-1" href="#__codelineno-40-1"></a>cat<span class="w"> </span>Pangenome/PanACoTA-pangenome_EsMa.log
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-41-1"><a id="__codelineno-41-1" name="__codelineno-41-1" href="#__codelineno-41-1"></a>[2024-08-25 18:54:07] :: INFO :: PanACoTA version 1.4.0
</span><span id="__span-41-2"><a id="__codelineno-41-2" name="__codelineno-41-2" href="#__codelineno-41-2"></a>[2024-08-25 18:54:07] :: INFO :: Command used
</span><span id="__span-41-3"><a id="__codelineno-41-3" name="__codelineno-41-3" href="#__codelineno-41-3"></a>     &gt; PanACoTA pangenome -l Annotation/LSTINFO-.lst -n EsMa -d Annotation/Proteins/ -o Pangenome -i 0.8
</span><span id="__span-41-4"><a id="__codelineno-41-4" name="__codelineno-41-4" href="#__codelineno-41-4"></a>[2024-08-25 18:54:07] :: INFO :: Building bank with all proteins to Annotation/Proteins/EsMa.All.prt
</span><span id="__span-41-5"><a id="__codelineno-41-5" name="__codelineno-41-5" href="#__codelineno-41-5"></a>[2024-08-25 18:54:08] :: INFO :: Will run MMseqs2 with:
</span><span id="__span-41-6"><a id="__codelineno-41-6" name="__codelineno-41-6" href="#__codelineno-41-6"></a>    - minimum sequence identity = 80.0%
</span><span id="__span-41-7"><a id="__codelineno-41-7" name="__codelineno-41-7" href="#__codelineno-41-7"></a>    - cluster mode 1
</span><span id="__span-41-8"><a id="__codelineno-41-8" name="__codelineno-41-8" href="#__codelineno-41-8"></a>[2024-08-25 18:54:08] :: INFO :: Creating database
</span><span id="__span-41-9"><a id="__codelineno-41-9" name="__codelineno-41-9" href="#__codelineno-41-9"></a>[2024-08-25 18:54:11] :: INFO :: Clustering proteins...
</span><span id="__span-41-10"><a id="__codelineno-41-10" name="__codelineno-41-10" href="#__codelineno-41-10"></a>[2024-08-25 18:54:36] :: INFO :: Converting mmseqs results to pangenome file
</span><span id="__span-41-11"><a id="__codelineno-41-11" name="__codelineno-41-11" href="#__codelineno-41-11"></a>[2024-08-25 18:54:36] :: INFO :: Pangenome has 9694 families.
</span><span id="__span-41-12"><a id="__codelineno-41-12" name="__codelineno-41-12" href="#__codelineno-41-12"></a>[2024-08-25 18:54:36] :: INFO :: Retrieving information from pan families
</span><span id="__span-41-13"><a id="__codelineno-41-13" name="__codelineno-41-13" href="#__codelineno-41-13"></a>[2024-08-25 18:54:36] :: INFO :: Generating qualitative and quantitative matrix, and summary file
</span><span id="__span-41-14"><a id="__codelineno-41-14" name="__codelineno-41-14" href="#__codelineno-41-14"></a>[2024-08-25 18:54:37] :: INFO :: DONE
</span></code></pre></div>
<details class="info" open="open">
<summary>Info</summary>
<p><code>[2024-08-25 18:54:36] :: INFO :: Pangenome has 9694 families.</code><br></p>
</details>
<p>That is, in this command, <code>PanACoTA</code> counted a total of <code>9694</code> protein families.</p>
<p>In this step, we took the annotated genomes, we took the proteins from them.<br>
And then the orthology table is constructed with them.<br>
The outcome of this step will be lists of genes grouped together.<br>
As output, we're going to see a bunch of logs, and one single table.</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-42-1"><a id="__codelineno-42-1" name="__codelineno-42-1" href="#__codelineno-42-1"></a>ls<span class="w"> </span>Pangenome/
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-43-1"><a id="__codelineno-43-1" name="__codelineno-43-1" href="#__codelineno-43-1"></a>PanACoTA-pangenome_EsMa.log
</span><span id="__span-43-2"><a id="__codelineno-43-2" name="__codelineno-43-2" href="#__codelineno-43-2"></a>PanACoTA-pangenome_EsMa.log.details
</span><span id="__span-43-3"><a id="__codelineno-43-3" name="__codelineno-43-3" href="#__codelineno-43-3"></a>PanACoTA-pangenome_EsMa.log.err
</span><span id="__span-43-4"><a id="__codelineno-43-4" name="__codelineno-43-4" href="#__codelineno-43-4"></a>PanGenome-EsMa.All.prt-clust-0.8-mode1.lst
</span><span id="__span-43-5"><a id="__codelineno-43-5" name="__codelineno-43-5" href="#__codelineno-43-5"></a>PanGenome-EsMa.All.prt-clust-0.8-mode1.lst.bin
</span><span id="__span-43-6"><a id="__codelineno-43-6" name="__codelineno-43-6" href="#__codelineno-43-6"></a>PanGenome-EsMa.All.prt-clust-0.8-mode1.lst.quali.txt
</span><span id="__span-43-7"><a id="__codelineno-43-7" name="__codelineno-43-7" href="#__codelineno-43-7"></a>PanGenome-EsMa.All.prt-clust-0.8-mode1.lst.quanti.txt
</span><span id="__span-43-8"><a id="__codelineno-43-8" name="__codelineno-43-8" href="#__codelineno-43-8"></a>PanGenome-EsMa.All.prt-clust-0.8-mode1.lst.summary.txt
</span><span id="__span-43-9"><a id="__codelineno-43-9" name="__codelineno-43-9" href="#__codelineno-43-9"></a>mmseq_EsMa.All.prt_0.8-mode1.log
</span><span id="__span-43-10"><a id="__codelineno-43-10" name="__codelineno-43-10" href="#__codelineno-43-10"></a>tmp_EsMa.All.prt_0.8-mode1
</span></code></pre></div>
<p>The only file of interest is <code>PanGenome-EsMa.All.prt-clust-0.8-mode1.lst</code>.<br>
Let's take a look at it.</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-44-1"><a id="__codelineno-44-1" name="__codelineno-44-1" href="#__codelineno-44-1"></a>head<span class="w"> </span>-10<span class="w"> </span>Pangenome/PanGenome-EsMa.All.prt-clust-0.8-mode1.lst
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-45-1"><a id="__codelineno-45-1" name="__codelineno-45-1" href="#__codelineno-45-1"></a>1 EsMa.0824.00001.0001i_04241 EsMa.0824.00002.0001i_00106 EsMa.0824.00003.0001i_00093 EsMa.0824.00004.0001i_00095 EsMa.0824.00005.0001i_00094 EsMa.0824.00006.0001i_03340 EsMa.0824.00007.0001i_02672 EsMa.0824.00008.0001i_00094 EsMa.0824.00009.0001i_00094 EsMa.0824.00010.0001i_04572 EsMa.0824.00011.0001i_02424 EsMa.0824.00012.0001i_02999 EsMa.0824.00013.0001i_02846 EsMa.0824.00014.0001i_00094
</span><span id="__span-45-2"><a id="__codelineno-45-2" name="__codelineno-45-2" href="#__codelineno-45-2"></a>2 EsMa.0824.00001.0001i_03873 EsMa.0824.00002.0001i_00130 EsMa.0824.00003.0001i_00117 EsMa.0824.00004.0001i_00119 EsMa.0824.00005.0001i_00118 EsMa.0824.00006.0001i_03364 EsMa.0824.00007.0001i_02648 EsMa.0824.00008.0001i_00118 EsMa.0824.00009.0001i_00126 EsMa.0824.00010.0001i_04604 EsMa.0824.00011.0001i_02400 EsMa.0824.00012.0001i_03032 EsMa.0824.00013.0001i_02821 EsMa.0824.00014.0001i_00126
</span><span id="__span-45-3"><a id="__codelineno-45-3" name="__codelineno-45-3" href="#__codelineno-45-3"></a>3 EsMa.0824.00008.0002i_04583 EsMa.0824.00010.0002i_04636 EsMa.0824.00012.0002i_04658
</span><span id="__span-45-4"><a id="__codelineno-45-4" name="__codelineno-45-4" href="#__codelineno-45-4"></a>4 EsMa.0824.00008.0002i_04649 EsMa.0824.00010.0002i_04668 EsMa.0824.00012.0002i_04736
</span><span id="__span-45-5"><a id="__codelineno-45-5" name="__codelineno-45-5" href="#__codelineno-45-5"></a>5 EsMa.0824.00007.0003i_04731 EsMa.0824.00008.0002i_04616 EsMa.0824.00010.0002i_04700 EsMa.0824.00012.0002i_04701 EsMa.0824.00012.0003i_04789
</span><span id="__span-45-6"><a id="__codelineno-45-6" name="__codelineno-45-6" href="#__codelineno-45-6"></a>6 EsMa.0824.00010.0003i_04833
</span><span id="__span-45-7"><a id="__codelineno-45-7" name="__codelineno-45-7" href="#__codelineno-45-7"></a>7 EsMa.0824.00001.0001i_02181 EsMa.0824.00002.0001i_02692 EsMa.0824.00003.0001i_02407 EsMa.0824.00004.0001i_02595 EsMa.0824.00005.0001i_02480 EsMa.0824.00006.0001i_00413 EsMa.0824.00007.0001i_00212 EsMa.0824.00008.0001i_02711 EsMa.0824.00009.0001i_02866 EsMa.0824.00010.0001i_02477 EsMa.0824.00011.0001i_00009 EsMa.0824.00011.0001i_04359 EsMa.0824.00012.0001i_01000 EsMa.0824.00013.0001i_01030 EsMa.0824.00014.0001i_02416
</span><span id="__span-45-8"><a id="__codelineno-45-8" name="__codelineno-45-8" href="#__codelineno-45-8"></a>8 EsMa.0824.00005.0001i_02448 EsMa.0824.00011.0001i_00041 EsMa.0824.00011.0001i_04391 EsMa.0824.00011.0001i_04451
</span><span id="__span-45-9"><a id="__codelineno-45-9" name="__codelineno-45-9" href="#__codelineno-45-9"></a>9 EsMa.0824.00002.0001i_03089 EsMa.0824.00003.0001i_02788 EsMa.0824.00005.0001i_02414 EsMa.0824.00008.0001i_02644 EsMa.0824.00008.0001i_02862 EsMa.0824.00011.0001i_00075 EsMa.0824.00011.0001i_04425 EsMa.0824.00011.0001i_04485 EsMa.0824.00014.0001i_02479
</span><span id="__span-45-10"><a id="__codelineno-45-10" name="__codelineno-45-10" href="#__codelineno-45-10"></a>10 EsMa.0824.00001.0001i_02131 EsMa.0824.00002.0001i_02651 EsMa.0824.00003.0001i_02366 EsMa.0824.00004.0001i_02431 EsMa.0824.00005.0001i_02379 EsMa.0824.00006.0001i_00454 EsMa.0824.00007.0001i_00253 EsMa.0824.00008.0001i_02607 EsMa.0824.00009.0001i_02686 EsMa.0824.00010.0001i_02436 EsMa.0824.00011.0001i_00110 EsMa.0824.00012.0001i_00951 EsMa.0824.00013.0001i_00978 EsMa.0824.00014.0001i_02375
</span></code></pre></div>
<p>In this file, each line corresponds to one orthologous row.<br>
The identifiers of all the genes that fall into this row are written in the line with a separator.<br>
For example, in line 6 we have one gene - it means that it is a singleton (it is so one, and no one has anything similar).<br>
Long rows are orthologous groups with many genes in them.<br></p>
<p>Further after this step, from the orthologous series, we identify the <code>core</code> - that is, the genes encoding proteins that occur in a single copy in each genome.<br></p>
<hr />
<h2 id="step-4-extract-pangenome"><strong>Step 4: Extract pangenome</strong></h2>
<p>Let's also break down the command below in parts:<br></p>
<ul>
<li><code>PanACoTA</code> - the name of the tool we are running.<br></li>
<li><code>corepers</code> - module of this tool that identifies and extracts <code>core</code>.<br></li>
<li>By key <code>-p</code> we pass the <code>PanGenome-EsMa.All.prt-clust-0.8-mode1.lst</code> file.<br></li>
<li>By key <code>-o</code> we specify where to store the output files.<br></li>
</ul>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-46-1"><a id="__codelineno-46-1" name="__codelineno-46-1" href="#__codelineno-46-1"></a>PanACoTA<span class="w"> </span>corepers<span class="w"> </span>-p<span class="w"> </span>Pangenome/PanGenome-EsMa.All.prt-clust-0.8-mode1.lst<span class="w"> </span>-o<span class="w"> </span>Coregenome
</span></code></pre></div>
<p>Let's look at the log file.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-47-1"><a id="__codelineno-47-1" name="__codelineno-47-1" href="#__codelineno-47-1"></a>cat<span class="w"> </span>Coregenome/PanACoTA-corepers.log
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-48-1"><a id="__codelineno-48-1" name="__codelineno-48-1" href="#__codelineno-48-1"></a>[2024-08-25 18:54:45] :: INFO :: PanACoTA version 1.4.0
</span><span id="__span-48-2"><a id="__codelineno-48-2" name="__codelineno-48-2" href="#__codelineno-48-2"></a>[2024-08-25 18:54:45] :: INFO :: Command used
</span><span id="__span-48-3"><a id="__codelineno-48-3" name="__codelineno-48-3" href="#__codelineno-48-3"></a>     &gt; PanACoTA corepers -p Pangenome/PanGenome-EsMa.All.prt-clust-0.8-mode1.lst -o Coregenome
</span><span id="__span-48-4"><a id="__codelineno-48-4" name="__codelineno-48-4" href="#__codelineno-48-4"></a>[2024-08-25 18:54:45] :: INFO :: Will generate a CoreGenome.
</span><span id="__span-48-5"><a id="__codelineno-48-5" name="__codelineno-48-5" href="#__codelineno-48-5"></a>[2024-08-25 18:54:45] :: INFO :: Retrieving info from binary file
</span><span id="__span-48-6"><a id="__codelineno-48-6" name="__codelineno-48-6" href="#__codelineno-48-6"></a>[2024-08-25 18:54:45] :: INFO :: Generating Persistent genome of a dataset containing 14 genomes
</span><span id="__span-48-7"><a id="__codelineno-48-7" name="__codelineno-48-7" href="#__codelineno-48-7"></a>[2024-08-25 18:54:45] :: INFO :: The core genome contains 2709 families, each one having exactly 14 members, from the 14 different genomes.
</span><span id="__span-48-8"><a id="__codelineno-48-8" name="__codelineno-48-8" href="#__codelineno-48-8"></a>[2024-08-25 18:54:45] :: INFO :: Persistent genome step done.
</span></code></pre></div>
<details class="info" open="open">
<summary>Info</summary>
<p><code>[2024-08-25 18:54:45] :: INFO :: The core genome contains 2709 families, each one having exactly 14 members, from the 14 different genomes.</code><br></p>
</details>
<p>That is, out of <code>9694</code> protein families, <code>2709</code> are single-copy and universal families.<br></p>
<p>Now let's take a look at the output.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-49-1"><a id="__codelineno-49-1" name="__codelineno-49-1" href="#__codelineno-49-1"></a>ls<span class="w"> </span>Coregenome/
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-50-1"><a id="__codelineno-50-1" name="__codelineno-50-1" href="#__codelineno-50-1"></a>PanACoTA-corepers.log
</span><span id="__span-50-2"><a id="__codelineno-50-2" name="__codelineno-50-2" href="#__codelineno-50-2"></a>PanACoTA-corepers.log.err
</span><span id="__span-50-3"><a id="__codelineno-50-3" name="__codelineno-50-3" href="#__codelineno-50-3"></a>PersGenome_PanGenome-EsMa.All.prt-clust-0.8-mode1.lst-all_1.lst
</span></code></pre></div>
<p>We only have logs and one table - <code>PersGenome_PanGenome-EsMa.All.prt-clust-0.8-mode1.lst-all_1.lst</code>.<br>
Let's take a look at it.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-51-1"><a id="__codelineno-51-1" name="__codelineno-51-1" href="#__codelineno-51-1"></a>head<span class="w"> </span>-10<span class="w"> </span>Coregenome/PersGenome_PanGenome-EsMa.All.prt-clust-0.8-mode1.lst-all_1.lst
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-52-1"><a id="__codelineno-52-1" name="__codelineno-52-1" href="#__codelineno-52-1"></a>1 EsMa.0824.00001.0001i_04241 EsMa.0824.00002.0001i_00106 EsMa.0824.00003.0001i_00093 EsMa.0824.00004.0001i_00095 EsMa.0824.00005.0001i_00094 EsMa.0824.00006.0001i_03340 EsMa.0824.00007.0001i_02672 EsMa.0824.00008.0001i_00094 EsMa.0824.00009.0001i_00094 EsMa.0824.00010.0001i_04572 EsMa.0824.00011.0001i_02424 EsMa.0824.00012.0001i_02999 EsMa.0824.00013.0001i_02846 EsMa.0824.00014.0001i_00094
</span><span id="__span-52-2"><a id="__codelineno-52-2" name="__codelineno-52-2" href="#__codelineno-52-2"></a>2 EsMa.0824.00001.0001i_03873 EsMa.0824.00002.0001i_00130 EsMa.0824.00003.0001i_00117 EsMa.0824.00004.0001i_00119 EsMa.0824.00005.0001i_00118 EsMa.0824.00006.0001i_03364 EsMa.0824.00007.0001i_02648 EsMa.0824.00008.0001i_00118 EsMa.0824.00009.0001i_00126 EsMa.0824.00010.0001i_04604 EsMa.0824.00011.0001i_02400 EsMa.0824.00012.0001i_03032 EsMa.0824.00013.0001i_02821 EsMa.0824.00014.0001i_00126
</span><span id="__span-52-3"><a id="__codelineno-52-3" name="__codelineno-52-3" href="#__codelineno-52-3"></a>10 EsMa.0824.00001.0001i_02131 EsMa.0824.00002.0001i_02651 EsMa.0824.00003.0001i_02366 EsMa.0824.00004.0001i_02431 EsMa.0824.00005.0001i_02379 EsMa.0824.00006.0001i_00454 EsMa.0824.00007.0001i_00253 EsMa.0824.00008.0001i_02607 EsMa.0824.00009.0001i_02686 EsMa.0824.00010.0001i_02436 EsMa.0824.00011.0001i_00110 EsMa.0824.00012.0001i_00951 EsMa.0824.00013.0001i_00978 EsMa.0824.00014.0001i_02375
</span><span id="__span-52-4"><a id="__codelineno-52-4" name="__codelineno-52-4" href="#__codelineno-52-4"></a>11 EsMa.0824.00001.0001i_02083 EsMa.0824.00002.0001i_02617 EsMa.0824.00003.0001i_02332 EsMa.0824.00004.0001i_02399 EsMa.0824.00005.0001i_02347 EsMa.0824.00006.0001i_00487 EsMa.0824.00007.0001i_00287 EsMa.0824.00008.0001i_02575 EsMa.0824.00009.0001i_02653 EsMa.0824.00010.0001i_02404 EsMa.0824.00011.0001i_00142 EsMa.0824.00012.0001i_00918 EsMa.0824.00013.0001i_00944 EsMa.0824.00014.0001i_02342
</span><span id="__span-52-5"><a id="__codelineno-52-5" name="__codelineno-52-5" href="#__codelineno-52-5"></a>12 EsMa.0824.00001.0001i_01323 EsMa.0824.00002.0001i_02588 EsMa.0824.00003.0001i_02303 EsMa.0824.00004.0001i_02370 EsMa.0824.00005.0001i_02318 EsMa.0824.00006.0001i_01160 EsMa.0824.00007.0001i_00316 EsMa.0824.00008.0001i_02546 EsMa.0824.00009.0001i_02534 EsMa.0824.00010.0001i_02375 EsMa.0824.00011.0001i_00174 EsMa.0824.00012.0001i_00889 EsMa.0824.00013.0001i_00915 EsMa.0824.00014.0001i_02313
</span><span id="__span-52-6"><a id="__codelineno-52-6" name="__codelineno-52-6" href="#__codelineno-52-6"></a>14 EsMa.0824.00001.0001i_01389 EsMa.0824.00002.0001i_02519 EsMa.0824.00003.0001i_02241 EsMa.0824.00004.0001i_02307 EsMa.0824.00005.0001i_02256 EsMa.0824.00006.0001i_01096 EsMa.0824.00007.0001i_00379 EsMa.0824.00008.0001i_02424 EsMa.0824.00009.0001i_02389 EsMa.0824.00010.0001i_02244 EsMa.0824.00011.0001i_00240 EsMa.0824.00012.0001i_00759 EsMa.0824.00013.0001i_00851 EsMa.0824.00014.0001i_02250
</span><span id="__span-52-7"><a id="__codelineno-52-7" name="__codelineno-52-7" href="#__codelineno-52-7"></a>15 EsMa.0824.00001.0001i_01420 EsMa.0824.00002.0001i_02486 EsMa.0824.00003.0001i_02209 EsMa.0824.00004.0001i_02275 EsMa.0824.00005.0001i_02224 EsMa.0824.00006.0001i_01062 EsMa.0824.00007.0001i_00411 EsMa.0824.00008.0001i_02391 EsMa.0824.00009.0001i_02356 EsMa.0824.00010.0001i_02211 EsMa.0824.00011.0001i_00272 EsMa.0824.00012.0001i_00727 EsMa.0824.00013.0001i_00819 EsMa.0824.00014.0001i_02217
</span><span id="__span-52-8"><a id="__codelineno-52-8" name="__codelineno-52-8" href="#__codelineno-52-8"></a>16 EsMa.0824.00001.0001i_01485 EsMa.0824.00002.0001i_02455 EsMa.0824.00003.0001i_02177 EsMa.0824.00004.0001i_02244 EsMa.0824.00005.0001i_02192 EsMa.0824.00006.0001i_01030 EsMa.0824.00007.0001i_00442 EsMa.0824.00008.0001i_02360 EsMa.0824.00009.0001i_02325 EsMa.0824.00010.0001i_02180 EsMa.0824.00011.0001i_00304 EsMa.0824.00012.0001i_00696 EsMa.0824.00013.0001i_00787 EsMa.0824.00014.0001i_02174
</span><span id="__span-52-9"><a id="__codelineno-52-9" name="__codelineno-52-9" href="#__codelineno-52-9"></a>20 EsMa.0824.00001.0001i_01791 EsMa.0824.00002.0001i_02262 EsMa.0824.00003.0001i_01989 EsMa.0824.00004.0001i_02098 EsMa.0824.00005.0001i_01999 EsMa.0824.00006.0001i_00822 EsMa.0824.00007.0001i_00574 EsMa.0824.00008.0001i_02152 EsMa.0824.00009.0001i_02124 EsMa.0824.00010.0001i_01983 EsMa.0824.00011.0001i_00496 EsMa.0824.00012.0001i_00561 EsMa.0824.00013.0001i_00575 EsMa.0824.00014.0001i_02021
</span><span id="__span-52-10"><a id="__codelineno-52-10" name="__codelineno-52-10" href="#__codelineno-52-10"></a>22 EsMa.0824.00001.0001i_01879 EsMa.0824.00002.0001i_02202 EsMa.0824.00003.0001i_01929 EsMa.0824.00004.0001i_02038 EsMa.0824.00005.0001i_01936 EsMa.0824.00006.0001i_00757 EsMa.0824.00007.0001i_00637 EsMa.0824.00008.0001i_02089 EsMa.0824.00009.0001i_02064 EsMa.0824.00010.0001i_01923 EsMa.0824.00011.0001i_00560 EsMa.0824.00012.0001i_00501 EsMa.0824.00013.0001i_00512 EsMa.0824.00014.0001i_01961
</span></code></pre></div>
<p>So here we are left with only those genes, those orthologues, which are in a single copy in all the organisms we have analysed.<br></p>
<p>At this step, it is possible to build more than just coregens.<br>
The <code>corepers</code> module has many more arguments, such as:<br></p>
<ul>
<li><code>-t</code> <tol>: % (between 0 and 1) of the persistent genome: a family is considered as persistent if it contains exactly one member in at least tol% of the genomes, and is absent in all other genomes. Default value for t is 1, meaning that all genomes must have a unique member. This corresponds to the coregenome (so no need to put this option if you want a coregenome). More relaxed definitions of a persistent genome can be used by using -X or -M options (see below).<br></li>
<li><code>-X</code>: add this option if you want to relax a little the definition of the persistent genome, to get a mixed persistent genome. With -X option, a family is considered as persistent if at least tol% (tol defined by -t <tol> parameter, see above) of the genomes have exactly one member in the family, but the other genomes can have either 0, either several members in the family. This is useful to add the families where, in some genomes, 1 protein has been split in several parts, because of sequencing or assembly error(s).<br></li>
<li><code>-M</code>: not compatible with -X. With this option, you get the multi persistent genome. It includes the strict and mixed persistent, but is even wider: the only condition for a family to be persistent is that it must have at least one member in at least tol% (tol still defined by -t <tol> parameter) of the genomes (independent of the copy number).<br></li>
</ul>
<hr />
<h2 id="step-5-aligning-the-sequence-of-core-gene-sequences"><strong>Step 5: Aligning the sequence of core gene sequences</strong></h2>
<details class="info" open="open">
<summary>Info</summary>
<p>For each of our objects, we have an amino acid sequence and a nucleotide sequence. That is, we can do alignments of both.<br>
For biological analysis, the protein sequence is preferred because:<br></p>
<ol>
<li>Larger alphabet = more letter variants - 20 vs. 4 for nucleotide sequences.<br></li>
<li>Amino acids have a more conservative structure (nucleotides are more variable, there are nonsynonymous substitutions, and amino acids are under stronger selection and it is easier for us to match conservative parts).<br></li>
</ol>
<p>However, when aligning amino acid sequences, we find it difficult to ‘go back’ and account for nucleotide substitutions.<br>
But there is a trick - we take the nucleotide sequences, translate them into amino acids and sort of align the amino acids, preserving the original triplets of nucleotides that were underneath those amino acids.<br></p>
<p>Then:<br></p>
<ul>
<li>We can account for honest correct reading frame shifts - like a single nucleotide dropout in a nucleotide alignment (i.e. we allow for a situation that is impossible in an amino acid alignment).<br></li>
<li>And at the same time we have information about the amino acid structure of the protein - a combination of two levels of information.<br></li>
</ul>
</details>
<p>Let's also break down the command below in parts:<br></p>
<ul>
<li><code>PanACoTA</code> - the name of the tool we are running.<br></li>
<li><code>align</code> - module of this tool that aligns sequences with <code>mafft</code>.<br></li>
<li>By key <code>-c</code> we pass the <code>PersGenome_PanGenome-EsMa.All.prt-clust-0.8-mode1.lst-all_1.lst</code> file.<br></li>
<li>By key <code>-l</code> we pass the <code>LSTINFO-.lst</code> file from the <strong>Step 2.2.</strong>.<br></li>
<li>By key <code>-n</code> we pass the four-character name for the data, which was set previously.<br></li>
<li>By key <code>-d</code> we pass the path to the output directory from <strong>Step 2.</strong>.<br></li>
<li>By key <code>-o</code> we specify where to store the output files.<br></li>
</ul>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-53-1"><a id="__codelineno-53-1" name="__codelineno-53-1" href="#__codelineno-53-1"></a>PanACoTA<span class="w"> </span>align<span class="w"> </span>-c<span class="w"> </span>Coregenome/PersGenome_PanGenome-EsMa.All.prt-clust-0.8-mode1.lst-all_1.lst<span class="w"> </span>-l<span class="w"> </span>Annotation/LSTINFO-.lst<span class="w"> </span>-n<span class="w"> </span>EsMa<span class="w"> </span>-d<span class="w"> </span>Annotation/<span class="w"> </span>-o<span class="w"> </span>Alignment
</span></code></pre></div>
<p>Let's go into the directory and look at the results.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-54-1"><a id="__codelineno-54-1" name="__codelineno-54-1" href="#__codelineno-54-1"></a>ls Alignment
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-55-1"><a id="__codelineno-55-1" name="__codelineno-55-1" href="#__codelineno-55-1"></a>Align-EsMa  PanACoTA-align_EsMa.log      PanACoTA-align_EsMa.log.err
</span><span id="__span-55-2"><a id="__codelineno-55-2" name="__codelineno-55-2" href="#__codelineno-55-2"></a>List-EsMa   PanACoTA-align_EsMa.log.details  Phylo-EsMa
</span></code></pre></div>
<p>We see a lot of log files.<br>
But now we are interested in the contents of the <code>Align-EsMa</code> directory.<br></p>
<h4 id="step-51-align-esma-directory"><strong>Step 5.1. <code>Align-EsMa</code> directory</strong></h4>
<p><strong><em>Input</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-56-1"><a id="__codelineno-56-1" name="__codelineno-56-1" href="#__codelineno-56-1"></a>ls -U Alignment/Align-EsMa | head -20
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-57-1"><a id="__codelineno-57-1" name="__codelineno-57-1" href="#__codelineno-57-1"></a>EsMa-complete.nucl.cat.aln
</span><span id="__span-57-2"><a id="__codelineno-57-2" name="__codelineno-57-2" href="#__codelineno-57-2"></a>EsMa-current.1.gen
</span><span id="__span-57-3"><a id="__codelineno-57-3" name="__codelineno-57-3" href="#__codelineno-57-3"></a>EsMa-current.1.miss.lst
</span><span id="__span-57-4"><a id="__codelineno-57-4" name="__codelineno-57-4" href="#__codelineno-57-4"></a>EsMa-current.1.prt
</span><span id="__span-57-5"><a id="__codelineno-57-5" name="__codelineno-57-5" href="#__codelineno-57-5"></a>EsMa-current.10.gen
</span><span id="__span-57-6"><a id="__codelineno-57-6" name="__codelineno-57-6" href="#__codelineno-57-6"></a>EsMa-current.10.miss.lst
</span><span id="__span-57-7"><a id="__codelineno-57-7" name="__codelineno-57-7" href="#__codelineno-57-7"></a>EsMa-current.10.prt
</span><span id="__span-57-8"><a id="__codelineno-57-8" name="__codelineno-57-8" href="#__codelineno-57-8"></a>EsMa-current.1000.gen
</span><span id="__span-57-9"><a id="__codelineno-57-9" name="__codelineno-57-9" href="#__codelineno-57-9"></a>EsMa-current.1000.miss.lst
</span><span id="__span-57-10"><a id="__codelineno-57-10" name="__codelineno-57-10" href="#__codelineno-57-10"></a>EsMa-current.1000.prt
</span><span id="__span-57-11"><a id="__codelineno-57-11" name="__codelineno-57-11" href="#__codelineno-57-11"></a>EsMa-current.1007.gen
</span><span id="__span-57-12"><a id="__codelineno-57-12" name="__codelineno-57-12" href="#__codelineno-57-12"></a>EsMa-current.1007.miss.lst
</span><span id="__span-57-13"><a id="__codelineno-57-13" name="__codelineno-57-13" href="#__codelineno-57-13"></a>EsMa-current.1007.prt
</span><span id="__span-57-14"><a id="__codelineno-57-14" name="__codelineno-57-14" href="#__codelineno-57-14"></a>EsMa-current.1009.gen
</span><span id="__span-57-15"><a id="__codelineno-57-15" name="__codelineno-57-15" href="#__codelineno-57-15"></a>EsMa-current.1009.miss.lst
</span><span id="__span-57-16"><a id="__codelineno-57-16" name="__codelineno-57-16" href="#__codelineno-57-16"></a>EsMa-current.1009.prt
</span><span id="__span-57-17"><a id="__codelineno-57-17" name="__codelineno-57-17" href="#__codelineno-57-17"></a>EsMa-current.1011.gen
</span><span id="__span-57-18"><a id="__codelineno-57-18" name="__codelineno-57-18" href="#__codelineno-57-18"></a>EsMa-current.1011.miss.lst
</span><span id="__span-57-19"><a id="__codelineno-57-19" name="__codelineno-57-19" href="#__codelineno-57-19"></a>EsMa-current.1011.prt
</span><span id="__span-57-20"><a id="__codelineno-57-20" name="__codelineno-57-20" href="#__codelineno-57-20"></a>EsMa-current.1012.gen
</span><span id="__span-57-21"><a id="__codelineno-57-21" name="__codelineno-57-21" href="#__codelineno-57-21"></a>ls: write error: Broken pipe
</span></code></pre></div>
<p>There are a lot of files in this folder (more than 1000), so we displayed only 20 of them on the screen.<br>
But now each separate file is not one organism (as it was in the previous steps), but one orthologous series (and it contains sequences from different organisms that were glued together).<br></p>
<ul>
<li><code>.prt</code> files correspond to protein, i.e. there will be amino acid sequences inside.<br></li>
<li><code>.gen</code> files correspond to nucleotides, so there will be nucleotide sequences inside.<br></li>
</ul>
<h3 id="step-52-phylo-esma-directory"><strong>Step 5.2. <code>Phylo-EsMa</code> directory</strong></h3>
<p>And now let's have a look at the contents of the <code>Phylo-EsMa</code> directory.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-58-1"><a id="__codelineno-58-1" name="__codelineno-58-1" href="#__codelineno-58-1"></a>ls Alignment/Phylo-EsMa/
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-59-1"><a id="__codelineno-59-1" name="__codelineno-59-1" href="#__codelineno-59-1"></a>EsMa.nucl.grp.aln
</span></code></pre></div>
<p><code>EsMa.nucl.grp.aln</code> is the alignment itselft, the main result of running the <code>PanACoTA</code>'s pipeline <code>align</code> module.<br>
It is the concatenate of the core genes alignment.<br>
It is from this alignment that we will build the phylogenetic tree in the next step.<br></p>
<hr />
<h2 id="step-6-build-a-phylogenomic-tree"><strong>Step 6: Build a phylogenomic tree</strong></h2>
<p>Let's also break down the command below in parts:<br></p>
<ul>
<li><code>PanACoTA</code> - the name of the tool we are running.<br></li>
<li><code>tree</code> - module of this tool for phylogenetic tree construction.<br></li>
<li>By key <code>-a</code> we pass the alignment file (<code>EsMa.nucl.grp.aln</code>).<br></li>
<li>By key <code>-s</code> we specify which tree building tool we want to use.<br></li>
<li>By key <code>-o</code> we specify where to store the output files.<br></li>
</ul>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-60-1"><a id="__codelineno-60-1" name="__codelineno-60-1" href="#__codelineno-60-1"></a>PanACoTA<span class="w"> </span>tree<span class="w"> </span>-a<span class="w"> </span>Alignment/Phylo-EsMa/EsMa.nucl.grp.aln<span class="w"> </span>-s<span class="w"> </span>iqtree2<span class="w"> </span>-o<span class="w"> </span>Tree
</span></code></pre></div>
<p>Let's look at the output!<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-61-1"><a id="__codelineno-61-1" name="__codelineno-61-1" href="#__codelineno-61-1"></a>ls<span class="w"> </span>Tree
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-62-1"><a id="__codelineno-62-1" name="__codelineno-62-1" href="#__codelineno-62-1"></a>EsMa.nucl.grp.aln.iqtree_tree.bionj   EsMa.nucl.grp.aln.iqtree_tree.treefile
</span><span id="__span-62-2"><a id="__codelineno-62-2" name="__codelineno-62-2" href="#__codelineno-62-2"></a>EsMa.nucl.grp.aln.iqtree_tree.ckp.gz  PanACoTA-tree-iqtree2.log
</span><span id="__span-62-3"><a id="__codelineno-62-3" name="__codelineno-62-3" href="#__codelineno-62-3"></a>EsMa.nucl.grp.aln.iqtree_tree.iqtree  PanACoTA-tree-iqtree2.log.details
</span><span id="__span-62-4"><a id="__codelineno-62-4" name="__codelineno-62-4" href="#__codelineno-62-4"></a>EsMa.nucl.grp.aln.iqtree_tree.log     PanACoTA-tree-iqtree2.log.err
</span><span id="__span-62-5"><a id="__codelineno-62-5" name="__codelineno-62-5" href="#__codelineno-62-5"></a>EsMa.nucl.grp.aln.iqtree_tree.mldist
</span></code></pre></div>
<ul>
<li><code>PanACoTA-tree-iqtree2.log</code>.<br></li>
<li><code>PanACoTA-tree-iqtree2.log.details</code>.<br></li>
<li><code>PanACoTA-tree-iqtree2.log.err</code>.<br></li>
</ul>
<p>It's all <code>PanACoTA</code>'s own logs</p>
<ul>
<li><code>EsMa.nucl.grp.aln.iqtree_tree.bionj</code> - this file contains the initial tree representation (tree draft).<br></li>
<li><code>EsMa.nucl.grp.aln.iqtree_tree.treefile</code> - the very tree file we are interested in.<br></li>
</ul>
<p>Let's take a look at it!<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-63-1"><a id="__codelineno-63-1" name="__codelineno-63-1" href="#__codelineno-63-1"></a>cat<span class="w"> </span>Tree/EsMa.nucl.grp.aln.iqtree_tree.treefile
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-64-1"><a id="__codelineno-64-1" name="__codelineno-64-1" href="#__codelineno-64-1"></a>(EsMa.0824.00001:0.0899533237,((EsMa.0824.00002:0.0014558718,(((EsMa.0824.00003:0.0012307785,EsMa.0824.00012:0.0012870353):0.0002683664,(EsMa.0824.00004:0.0013314298,(EsMa.0824.00006:0.0026908476,((EsMa.0824.00009:0.0000384090,EsMa.0824.00010:0.0000274042):0.0015596552,EsMa.0824.00013:0.0014403274):0.0002421241):0.0003947757):0.0002574774):0.0000880096,(EsMa.0824.00007:0.0012606069,EsMa.0824.00008:0.0011876549):0.0002556339):0.0001349309):0.0002322047,EsMa.0824.00014:0.0012843745):0.0012230243,(EsMa.0824.00005:0.0000163401,EsMa.0824.00011:0.0001690095):0.0009982711);
</span></code></pre></div>
<p>This is a tree in <code>Newick</code> format.<br></p>
<h3 id="step-61-tree-visualization"><strong>Step 6.1. Tree visualization</strong></h3>
<p>For more details please visit <a href="https://github.com/iliapopov17/NGS-Handbook/tree/main/04_Phylogenetics">Phylogenetics handbook</a>.<br>
Here we will not use <code>iTOL</code> or <code>FigTree</code> but we will visualize the tree in the fastest way possible - using pseudo-graphics.<br></p>
<p>Import <code>Phylo</code> module from <code>Biopython</code>.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-65-1"><a id="__codelineno-65-1" name="__codelineno-65-1" href="#__codelineno-65-1"></a><span class="kn">from</span><span class="w"> </span><span class="nn">Bio</span><span class="w"> </span><span class="kn">import</span> <span class="n">Phylo</span>
</span></code></pre></div>
<p>Read the <code>EsMa.nucl.grp.aln.iqtree_tree.treefile</code> file to the <code>tree</code> variable.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-66-1"><a id="__codelineno-66-1" name="__codelineno-66-1" href="#__codelineno-66-1"></a><span class="n">tree</span> <span class="o">=</span> <span class="n">Phylo</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;Tree/EsMa.nucl.grp.aln.iqtree_tree.treefile&quot;</span><span class="p">,</span> <span class="s2">&quot;newick&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p>Visualize the tree!<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-67-1"><a id="__codelineno-67-1" name="__codelineno-67-1" href="#__codelineno-67-1"></a><span class="n">Phylo</span><span class="o">.</span><span class="n">draw_ascii</span><span class="p">(</span><span class="n">tree</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-68-1"><a id="__codelineno-68-1" name="__codelineno-68-1" href="#__codelineno-68-1"></a>  __________________________________________________________ EsMa.0824.00001
</span><span id="__span-68-2"><a id="__codelineno-68-2" name="__codelineno-68-2" href="#__codelineno-68-2"></a> |
</span><span id="__span-68-3"><a id="__codelineno-68-3" name="__codelineno-68-3" href="#__codelineno-68-3"></a> , EsMa.0824.00002
</span><span id="__span-68-4"><a id="__codelineno-68-4" name="__codelineno-68-4" href="#__codelineno-68-4"></a> |
</span><span id="__span-68-5"><a id="__codelineno-68-5" name="__codelineno-68-5" href="#__codelineno-68-5"></a> |, EsMa.0824.00003
</span><span id="__span-68-6"><a id="__codelineno-68-6" name="__codelineno-68-6" href="#__codelineno-68-6"></a> ||
</span><span id="__span-68-7"><a id="__codelineno-68-7" name="__codelineno-68-7" href="#__codelineno-68-7"></a> || EsMa.0824.00012
</span><span id="__span-68-8"><a id="__codelineno-68-8" name="__codelineno-68-8" href="#__codelineno-68-8"></a> ||
</span><span id="__span-68-9"><a id="__codelineno-68-9" name="__codelineno-68-9" href="#__codelineno-68-9"></a> |, EsMa.0824.00004
</span><span id="__span-68-10"><a id="__codelineno-68-10" name="__codelineno-68-10" href="#__codelineno-68-10"></a> ||
</span><span id="__span-68-11"><a id="__codelineno-68-11" name="__codelineno-68-11" href="#__codelineno-68-11"></a> ||_ EsMa.0824.00006
</span><span id="__span-68-12"><a id="__codelineno-68-12" name="__codelineno-68-12" href="#__codelineno-68-12"></a> ||
</span><span id="__span-68-13"><a id="__codelineno-68-13" name="__codelineno-68-13" href="#__codelineno-68-13"></a>_||, EsMa.0824.00009
</span><span id="__span-68-14"><a id="__codelineno-68-14" name="__codelineno-68-14" href="#__codelineno-68-14"></a> |,|
</span><span id="__span-68-15"><a id="__codelineno-68-15" name="__codelineno-68-15" href="#__codelineno-68-15"></a> ||| EsMa.0824.00010
</span><span id="__span-68-16"><a id="__codelineno-68-16" name="__codelineno-68-16" href="#__codelineno-68-16"></a> ||
</span><span id="__span-68-17"><a id="__codelineno-68-17" name="__codelineno-68-17" href="#__codelineno-68-17"></a> || EsMa.0824.00013
</span><span id="__span-68-18"><a id="__codelineno-68-18" name="__codelineno-68-18" href="#__codelineno-68-18"></a> ||
</span><span id="__span-68-19"><a id="__codelineno-68-19" name="__codelineno-68-19" href="#__codelineno-68-19"></a> |, EsMa.0824.00007
</span><span id="__span-68-20"><a id="__codelineno-68-20" name="__codelineno-68-20" href="#__codelineno-68-20"></a> ||
</span><span id="__span-68-21"><a id="__codelineno-68-21" name="__codelineno-68-21" href="#__codelineno-68-21"></a> || EsMa.0824.00008
</span><span id="__span-68-22"><a id="__codelineno-68-22" name="__codelineno-68-22" href="#__codelineno-68-22"></a> |
</span><span id="__span-68-23"><a id="__codelineno-68-23" name="__codelineno-68-23" href="#__codelineno-68-23"></a> | EsMa.0824.00014
</span><span id="__span-68-24"><a id="__codelineno-68-24" name="__codelineno-68-24" href="#__codelineno-68-24"></a> |
</span><span id="__span-68-25"><a id="__codelineno-68-25" name="__codelineno-68-25" href="#__codelineno-68-25"></a> , EsMa.0824.00005
</span><span id="__span-68-26"><a id="__codelineno-68-26" name="__codelineno-68-26" href="#__codelineno-68-26"></a> |
</span><span id="__span-68-27"><a id="__codelineno-68-27" name="__codelineno-68-27" href="#__codelineno-68-27"></a> | EsMa.0824.00011
</span></code></pre></div>
<p>That's right! The longest branch is <em>Shigella flexneri</em>.<br></p>









  




                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../02_Variant_Calling/" class="md-footer__link md-footer__link--prev" aria-label="Previous: II. Genomic Variation Analysis">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                II. Genomic Variation Analysis
              </div>
            </div>
          </a>
        
        
          
          <a href="../IV.%20Phylogenetics/04_00_Overview/" class="md-footer__link md-footer__link--next" aria-label="Next: Overview">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Overview
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2025 <a href="https://iliapopov17.github.io/">Ilia Popov</a>. Funded by RSF grant No. <a href="https://rscf.ru/en/project/25-24-00351/">25-24-00351</a>.
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://iliapopov17.github.io/" target="_blank" rel="noopener" title="iliapopov17.github.io" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 6.7.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M64 32C28.7 32 0 60.7 0 96v320c0 35.3 28.7 64 64 64h448c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64zm80 256h64c44.2 0 80 35.8 80 80 0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16 0-44.2 35.8-80 80-80m-32-96a64 64 0 1 1 128 0 64 64 0 1 1-128 0m256-32h128c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16m0 64h128c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16m0 64h128c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16"/></svg>
    </a>
  
    
    
    
    
    <a href="mailto:<iljapopov17@gmail.com>" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.7.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4l217.6 163.2c11.4 8.5 27 8.5 38.4 0l217.6-163.2c12.1-9.1 19.2-23.3 19.2-38.4 0-26.5-21.5-48-48-48zM0 176v208c0 35.3 28.7 64 64 64h384c35.3 0 64-28.7 64-64V176L294.4 339.2a63.9 63.9 0 0 1-76.8 0z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://github.com/iliapopov17" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.7.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://rscf.ru/en/project/25-24-00351/" target="_blank" rel="noopener" title="rscf.ru" class="md-social__link">
      <?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by Pixelmator Pro 3.6.12 -->
<svg width="1785" height="1341" viewBox="0 0 1785 1341" xmlns="http://www.w3.org/2000/svg">
    <g id="-1">
        <g id="Group">
            <g id="g1">
                <g id="g2">
                    <path id="Path" d="M 126.074524 926.216187 C 126.074524 930.877441 125.486938 935.489136 124.325317 940.066162 C 123.162476 944.64209 121.371094 948.891724 118.96228 952.795532 C 116.546082 956.706909 113.507568 960.199097 109.851685 963.274902 C 106.189575 966.361816 101.86792 968.728394 96.873108 970.393311 L 96.873108 970.641235 C 104.022522 972.971924 109.179749 976.881958 112.345947 982.366455 C 115.505981 987.862183 117.999084 993.934326 119.833801 1000.592773 L 133.562439 1048.505127 L 96.873108 1048.505127 L 83.894531 1003.582886 C 81.231689 994.104004 77.527405 987.693604 72.787964 984.369873 C 68.047302 981.04126 62.18219 979.376221 55.190247 979.376221 L 36.722229 979.376221 L 36.722229 1048.505127 L 0.287048 1048.505127 L 0.287048 877.04541 L 69.91803 877.04541 C 79.736511 877.04541 88.175293 878.340942 95.251587 880.91333 C 102.321655 883.498047 108.144592 887.032471 112.720398 891.525146 C 117.291199 896.016724 120.663208 901.258301 122.8302 907.244751 C 124.991028 913.237549 126.074524 919.563843 126.074524 926.216187 Z M 62.926025 954.17041 C 65.922424 954.17041 68.918823 953.758911 71.915222 952.929443 C 74.906616 952.100098 77.653809 950.695435 80.154358 948.70459 C 82.648621 946.718506 84.681763 944.072998 86.26239 940.755493 C 87.847961 937.445435 88.635193 933.30481 88.635193 928.340942 C 88.635193 919.732422 86.218994 913.39502 81.401489 909.339966 C 76.570374 905.289795 70.32959 903.255371 62.678101 903.255371 L 36.722229 903.255371 L 36.722229 954.17041 L 62.926025 954.17041 Z"/>
                    <path id="path1" d="M 190.185059 1051.252441 C 182.031494 1051.252441 175.499512 1050.168823 170.596313 1048.008057 C 165.680908 1045.84729 161.897339 1042.85083 159.240601 1039.025146 C 156.576416 1035.199341 154.827271 1030.787231 153.997803 1025.798584 C 153.163452 1020.805054 152.750732 1015.398682 152.750732 1009.575684 L 152.750732 925.968262 L 188.442139 925.968262 L 188.937988 1010.320801 C 189.270142 1012.983643 190.015259 1015.314453 191.182983 1017.311523 C 192.345825 1019.303711 194.010864 1020.895508 196.177979 1022.057129 C 198.338623 1023.219971 201.328857 1023.801514 205.160767 1023.801514 C 208.653076 1023.801514 212.770142 1022.886475 217.515869 1021.052979 C 222.255249 1019.224365 226.87439 1016.313599 231.365845 1012.317993 L 231.365845 925.968262 L 267.055908 925.720215 L 267.055908 1048.505127 L 233.865234 1048.505127 L 232.860962 1035.278687 L 231.861816 1034.781616 C 228.036011 1038.11145 224.168091 1040.810303 220.258057 1042.892944 C 216.346802 1044.975708 212.564453 1046.634399 208.901001 1047.881592 C 205.238892 1049.133667 201.788818 1050.005249 198.544434 1050.502441 L 190.185059 1051.252441 Z"/>
                    <path id="path2" d="M 341.753662 922.965698 C 348.24231 922.965698 355.43396 923.679688 363.3396 925.090454 C 371.239014 926.506226 378.441772 928.958374 384.925415 932.456909 L 377.189697 954.418335 C 370.536133 951.917847 364.379761 950.216919 358.720459 949.297119 C 353.067261 948.383545 347.824585 947.929688 342.999634 947.929688 C 338.508057 947.929688 334.392212 948.637573 330.645874 950.053345 C 326.904419 951.464111 325.033691 954.502686 325.033691 959.157837 C 325.033691 960.822754 325.736572 962.275757 327.152344 963.522827 C 328.56311 964.77002 330.312378 965.89563 332.39502 966.893555 L 362.468018 976.506348 C 367.78894 978.419312 372.654785 980.955688 377.068115 984.121826 C 381.475342 987.281982 385.093994 991.227905 387.921875 995.973633 C 390.754639 1000.714233 392.165405 1006.664795 392.165405 1013.81311 C 392.165405 1025.628662 387.424683 1034.703491 377.939697 1041.022217 C 368.45459 1047.342285 354.144653 1050.502441 335.015869 1050.502441 L 323.405884 1050.005249 L 310.305908 1048.255981 L 297.079346 1045.139404 C 292.751587 1043.892212 288.757202 1042.35498 285.099976 1040.520142 L 292.837036 1018.310669 C 302.981689 1021.972778 311.716675 1024.255127 319.040894 1025.168823 C 326.360229 1026.088623 332.097534 1026.543701 336.262939 1026.543701 C 342.085938 1026.543701 346.995117 1025.58667 350.984619 1023.673706 C 354.979004 1021.760864 356.977295 1018.22644 356.977295 1013.069214 C 356.977295 1010.574829 356.311646 1008.534424 354.979004 1006.954956 C 353.647461 1005.369385 351.862305 1004.000732 349.615967 1002.833008 L 317.545898 991.979248 C 312.21875 990.066284 307.515259 987.777832 303.441528 985.11499 C 299.361694 982.450806 296.081421 979.128296 293.580933 975.132813 C 291.086548 971.137207 289.840698 966.143555 289.840698 960.157104 C 289.840698 948.341309 294.246582 939.194702 303.06604 932.704834 C 311.886475 926.216187 324.779541 922.965698 341.753662 922.965698 Z"/>
                    <path id="path3" d="M 460.203369 922.965698 C 466.693237 922.965698 473.884766 923.679688 481.790405 925.090454 C 489.689941 926.506226 496.892578 928.958374 503.376343 932.456909 L 495.639282 954.418335 C 488.986938 951.917847 482.830566 950.216919 477.171387 949.297119 C 471.516968 948.383545 466.275391 947.929688 461.450439 947.929688 C 456.958984 947.929688 452.836914 948.637573 449.095459 950.053345 C 445.355225 951.464111 443.484497 954.502686 443.484497 959.157837 C 443.484497 960.822754 444.186279 962.275757 445.603149 963.522827 C 447.014038 964.77002 448.763184 965.89563 450.845947 966.893555 L 480.917725 976.506348 C 486.238525 978.419312 491.105713 980.955688 495.519043 984.121826 C 499.925049 987.281982 503.538696 991.227905 506.372681 995.973633 C 509.199341 1000.714233 510.61499 1006.664795 510.61499 1013.81311 C 510.61499 1025.628662 505.87561 1034.703491 496.390503 1041.022217 C 486.904297 1047.342285 472.595459 1050.502441 453.460571 1050.502441 L 441.856812 1050.005249 L 428.756714 1048.255981 L 415.524048 1045.139404 C 411.202393 1043.892212 407.206787 1042.35498 403.550903 1040.520142 L 411.286743 1018.310669 C 421.432495 1021.972778 430.16748 1024.255127 437.491699 1025.168823 C 444.809814 1026.088623 450.548462 1026.543701 454.713867 1026.543701 C 460.536743 1026.543701 465.439819 1025.58667 469.435547 1023.673706 C 473.42981 1021.760864 475.428223 1018.22644 475.428223 1013.069214 C 475.428223 1010.574829 474.762451 1008.534424 473.42981 1006.954956 C 472.098389 1005.369385 470.31311 1004.000732 468.060669 1002.833008 L 435.996704 991.979248 C 430.663452 990.066284 425.966187 987.777832 421.892456 985.11499 C 417.8125 982.450806 414.532349 979.128296 412.031738 975.132813 C 409.537476 971.137207 408.290283 966.143555 408.290283 960.157104 C 408.290283 948.341309 412.69751 939.194702 421.516846 932.704834 C 430.337402 926.216187 443.230347 922.965698 460.203369 922.965698 Z"/>
                    <path id="path4" d="M 570.439819 892.021118 C 570.439819 897.348145 568.648438 901.845825 565.070679 905.501709 C 561.494019 909.163818 556.215454 910.992432 549.223511 910.992432 C 542.571045 910.992432 537.498169 909.163818 533.999756 905.501709 C 530.506226 901.845825 528.756958 897.348145 528.756958 892.021118 C 528.756958 886.536621 530.549683 881.960815 534.126221 878.298706 C 537.703979 874.635376 542.734741 872.808105 549.223511 872.808105 C 556.215454 872.808105 561.494019 874.635376 565.070679 878.298706 C 568.648438 881.960815 570.439819 886.536621 570.439819 892.021118 Z M 531.505371 1048.505127 L 531.505371 925.968262 L 567.195557 925.968262 L 567.195557 1048.505127 L 531.505371 1048.505127 Z"/>
                    <path id="path5" d="M 701.064697 1024.799438 L 717.038452 1024.799438 L 718.787598 1044.763672 C 714.793335 1046.597168 710.131958 1048.092285 704.811157 1049.255127 C 699.484131 1050.416748 694.156982 1050.998291 688.836182 1050.998291 C 683.509155 1050.998291 679.059814 1050.090698 675.483276 1048.255981 C 671.905518 1046.428711 669.780518 1042.439331 669.121094 1036.276611 L 668.619019 1035.774536 C 664.289917 1039.606567 659.968262 1042.517334 655.640503 1044.509521 C 651.312622 1046.513062 647.281006 1048.008057 643.539551 1049.007324 C 639.793213 1050.005249 636.385132 1050.58667 633.303223 1050.750366 L 626.19104 1050.998291 C 618.539551 1050.998291 612.256592 1049.799561 607.347412 1047.384399 C 602.43811 1044.975708 598.57019 1041.97937 595.742432 1038.401611 C 592.909668 1034.823608 590.960815 1030.955811 589.883545 1026.791504 C 588.793823 1022.638672 588.254517 1018.806641 588.254517 1015.314453 C 588.254517 1008.98938 589.417358 1003.461548 591.754272 998.715698 C 594.078735 993.970093 597.280884 989.980713 601.360718 986.73645 C 605.434448 983.492188 610.22229 980.955688 615.706787 979.128296 C 621.203613 977.293579 627.190186 976.216309 633.678833 975.877808 L 667.123901 973.88562 L 667.123901 970.88916 C 667.123901 963.903442 665.628784 958.619751 662.632446 955.04187 C 659.635986 951.464111 653.557739 949.672729 644.411133 949.672729 C 638.418213 949.672729 631.887573 950.259155 624.822388 951.421997 C 617.746094 952.589844 611.299561 954.084961 605.476685 955.913452 L 599.738037 933.703979 C 607.056152 930.543945 615.543091 928.044678 625.191772 926.216187 C 634.841797 924.387573 644.078857 923.473877 652.898193 923.473877 C 660.216187 923.473877 666.874756 924.218872 672.861206 925.720215 C 678.854004 927.215332 683.92688 929.630371 688.086182 932.952759 C 692.244141 936.282593 695.446289 940.647705 697.698975 946.05896 C 699.94397 951.464111 701.064697 958.15979 701.064697 966.143555 L 701.064697 1024.799438 Z M 642.41394 995.599121 C 630.93042 996.597168 625.191772 1002.172119 625.191772 1012.317993 C 625.191772 1017.47522 626.524536 1021.391479 629.187378 1024.049438 C 631.845337 1026.712158 636.087646 1028.038696 641.917969 1028.038696 C 645.742554 1028.038696 649.901733 1027.087891 654.393311 1025.168823 C 658.884644 1023.262207 663.128296 1020.550781 667.123901 1017.063599 L 667.123901 993.600708 L 642.41394 995.599121 Z"/>
                    <path id="path6" d="M 815.47168 1048.505127 L 814.968384 963.903442 C 814.636108 961.240601 813.842773 958.909912 812.584351 956.912598 C 811.331055 954.920532 809.527222 953.377075 807.19043 952.299683 C 804.84729 951.216187 801.748047 950.670654 797.904907 950.670654 C 793.722046 950.670654 789.042236 951.591797 783.855225 953.419189 C 778.666992 955.253906 773.975952 957.414795 769.799316 959.909058 L 769.799316 1048.505127 L 734.109375 1048.505127 L 734.109375 925.968262 L 767.807251 925.968262 L 768.298218 937.196289 L 769.302368 937.693481 C 776.47644 932.86853 784.030151 929.206421 791.954346 926.712036 C 799.883423 924.218872 806.766479 922.965698 812.60791 922.965698 C 820.786377 922.965698 827.37793 924.091309 832.383789 926.337646 C 837.389771 928.588867 841.270142 931.621338 844.02478 935.447144 C 846.779419 939.278931 848.655029 943.770508 849.6604 948.927734 C 850.658447 954.084961 851.161743 959.491211 851.161743 965.150635 L 851.161743 1048.505127 L 815.47168 1048.505127 Z"/>
                    <path id="path7" d="M 1014.624268 873.55188 C 1023.111328 873.55188 1032.178589 874.429565 1041.828491 876.173828 C 1051.477173 877.923096 1060.550659 881.125244 1069.031494 885.786621 L 1059.551392 910.743286 C 1051.065552 907.244751 1042.784302 904.962402 1034.714966 903.872803 C 1026.645752 902.79541 1019.533447 902.257446 1013.377075 902.257446 C 1005.556885 902.257446 999.400513 904.084839 994.908936 907.741943 C 990.417603 911.404053 988.171265 916.899658 988.171265 924.218872 C 988.171265 927.547607 989.128174 930.331909 991.041138 932.578369 C 992.954102 934.82959 995.490479 936.778564 998.655396 938.443359 C 1001.809204 940.108398 1005.429321 941.525269 1009.509155 942.687012 L 1041.204956 951.174072 C 1047.947754 953.164917 1054.182129 956.083374 1059.927002 959.909058 C 1065.665649 963.739868 1070.405029 968.812744 1074.15271 975.132813 C 1077.892944 981.457764 1079.763672 989.690674 1079.763672 999.842773 C 1079.763672 1016.149902 1073.7771 1028.752808 1061.797729 1037.651489 C 1049.818359 1046.555054 1031.760742 1050.998291 1007.638428 1050.998291 L 993.286255 1050.254517 C 988.0448 1049.751221 982.681763 1048.965088 977.19104 1047.881592 C 971.700317 1046.803101 966.209595 1045.387329 960.720093 1043.644287 C 955.229614 1041.893799 949.988037 1039.775146 944.993164 1037.282104 L 954.47937 1011.318726 C 966.621216 1016.149902 977.523193 1019.224365 987.173096 1020.550781 C 996.821899 1021.88855 1004.557739 1022.554321 1010.386841 1022.554321 C 1020.532715 1022.554321 1028.184204 1020.84729 1033.347534 1017.432983 C 1038.498535 1014.025024 1041.083374 1008.164917 1041.083374 999.842773 C 1041.083374 996.179321 1040.417725 993.147095 1039.086182 990.732056 C 1037.75354 988.323242 1035.840698 986.234375 1033.347534 984.491333 C 1030.847168 982.748291 1027.856934 981.289185 1024.364502 980.126343 L 988.668213 970.0177 C 981.016724 967.771362 974.278931 964.818359 968.455933 961.15625 C 962.626831 957.499146 958.013916 952.881104 954.605957 947.306152 C 951.191772 941.731201 949.490845 934.533325 949.490845 925.720215 C 949.490845 910.077637 954.933228 897.474609 965.835205 887.905273 C 976.731079 878.340942 992.996216 873.55188 1014.624268 873.55188 Z"/>
                    <path id="path8" d="M 1133.668823 987.112061 C 1133.668823 998.431885 1135.79248 1007.287231 1140.036133 1013.691528 C 1144.279541 1020.095825 1151.973267 1023.304321 1163.123291 1023.304321 C 1168.941162 1023.304321 1174.516357 1022.718018 1179.843262 1021.548828 C 1185.17041 1020.387329 1190.575439 1018.892212 1196.066162 1017.063599 L 1202.802734 1040.768066 C 1196.64624 1043.934448 1189.53418 1046.349365 1181.464844 1048.008057 C 1173.390625 1049.666748 1165.115479 1050.502441 1156.634644 1050.502441 C 1144.81897 1050.502441 1135.126831 1048.885742 1127.554688 1045.635254 C 1119.982422 1042.397095 1113.952515 1037.941528 1109.46106 1032.287231 C 1104.969482 1026.62793 1101.852905 1020.053833 1100.102417 1012.565918 C 1098.359375 1005.078003 1097.481567 997.094238 1097.481567 988.607178 C 1097.481567 981.457764 1098.317261 974.013306 1099.975952 966.276245 C 1101.640869 958.534302 1104.721558 951.464111 1109.213135 945.059692 C 1113.704468 938.655396 1119.860962 933.413818 1127.68103 929.333984 C 1135.496338 925.259033 1145.648315 923.219727 1158.129761 923.219727 C 1163.450562 923.219727 1169.401123 923.595337 1175.975342 924.345459 C 1182.54834 925.090454 1189.328369 926.712036 1196.313965 929.206421 L 1190.823486 954.17041 C 1185.496338 952.669189 1180.62915 951.712036 1176.223389 951.300415 L 1164.122437 950.670654 C 1154.140259 950.670654 1146.56189 953.879028 1141.410889 960.283447 C 1136.253662 966.687866 1133.668823 975.62854 1133.668823 987.112061 Z"/>
                    <path id="path9" d="M 1260.479248 892.021118 C 1260.479248 897.348145 1258.686768 901.845825 1255.110107 905.501709 C 1251.532227 909.163818 1246.253418 910.992432 1239.261475 910.992432 C 1232.609131 910.992432 1227.536133 909.163818 1224.037842 905.501709 C 1220.54541 901.845825 1218.796143 897.348145 1218.796143 892.021118 C 1218.796143 886.536621 1220.587646 881.960815 1224.165527 878.298706 C 1227.742188 874.635376 1232.772705 872.808105 1239.261475 872.808105 C 1246.253418 872.808105 1251.532227 874.635376 1255.110107 878.298706 C 1258.686768 881.960815 1260.479248 886.536621 1260.479248 892.021118 Z M 1221.543457 1048.505127 L 1221.543457 925.968262 L 1257.233643 925.968262 L 1257.233643 1048.505127 L 1221.543457 1048.505127 Z"/>
                    <path id="path10" d="M 1340.260986 922.717651 C 1353.40918 922.717651 1363.723633 924.80542 1371.21167 928.958374 C 1378.699463 933.116455 1384.31665 938.279785 1388.058105 944.429932 C 1391.804443 950.592529 1394.128906 957.124634 1395.048828 964.024902 C 1395.95752 970.931396 1396.41748 977.129883 1396.41748 982.620605 L 1396.41748 995.351196 L 1316.054199 995.351196 C 1316.054199 1004.914429 1318.427002 1012.130737 1323.166504 1016.996704 C 1327.912109 1021.86377 1335.769531 1024.297241 1346.749512 1024.297241 C 1353.40918 1024.297241 1360.061523 1023.595581 1366.720215 1022.178711 C 1373.372559 1020.767944 1379.861084 1018.892212 1386.187256 1016.561523 L 1393.420898 1038.395386 C 1385.267578 1042.838501 1376.702393 1045.968628 1367.719238 1047.783691 C 1358.730225 1049.593628 1350.001465 1050.502441 1341.51416 1050.502441 C 1329.195313 1050.502441 1319.067871 1048.795288 1311.120117 1045.387329 C 1303.172363 1041.97937 1296.925537 1037.366455 1292.385742 1031.537231 C 1287.8396 1025.714233 1284.704346 1019.018555 1282.966309 1011.446411 C 1281.229492 1003.874268 1280.370361 995.847046 1280.370361 987.359985 C 1280.370361 980.707764 1281.198486 973.553345 1282.863525 965.89563 C 1284.522217 958.244141 1287.604004 951.258423 1292.095459 944.93335 C 1296.587158 938.613281 1302.706299 933.328369 1310.442139 929.084839 C 1318.179199 924.842529 1328.11792 922.717651 1340.260986 922.717651 Z M 1341.646973 945.429321 C 1337.131836 945.429321 1333.366943 946.185425 1330.358154 947.680542 C 1327.343018 949.175659 1324.879883 951.216187 1322.95459 953.7948 C 1321.029297 956.379639 1319.565186 959.369873 1318.559814 962.77771 C 1317.555664 966.191895 1316.889893 969.727661 1316.556396 973.383545 L 1363.977783 973.383545 C 1363.802002 965.236084 1362.174072 958.534302 1359.081055 953.292725 C 1355.981689 948.049927 1350.169922 945.429321 1341.646973 945.429321 Z"/>
                    <path id="path11" d="M 1498.445801 1048.505127 L 1497.942627 963.903442 C 1497.604248 961.240601 1496.810791 958.909912 1495.558594 956.912598 C 1494.305176 954.920532 1492.501465 953.377075 1490.158447 952.299683 C 1487.81665 951.216187 1484.722168 950.670654 1480.873047 950.670654 C 1476.696289 950.670654 1472.011475 951.591797 1466.829346 953.419189 C 1461.641357 955.253906 1456.950195 957.414795 1452.773682 959.909058 L 1452.773682 1048.505127 L 1417.083496 1048.505127 L 1417.083496 925.968262 L 1450.78125 925.968262 L 1451.272217 937.196289 L 1452.277588 937.693481 C 1459.450684 932.86853 1467.00415 929.206421 1474.928467 926.712036 C 1482.858887 924.218872 1489.740479 922.965698 1495.582031 922.965698 C 1503.760498 922.965698 1510.352051 924.091309 1515.358154 926.337646 C 1520.364014 928.588867 1524.244141 931.621338 1526.999023 935.447144 C 1529.753662 939.278931 1531.62915 943.770508 1532.634766 948.927734 C 1533.627686 954.084961 1534.135986 959.491211 1534.135986 965.150635 L 1534.135986 1048.505127 L 1498.445801 1048.505127 Z"/>
                    <path id="path12" d="M 1590.48584 987.112061 C 1590.48584 998.431885 1592.604492 1007.287231 1596.848145 1013.691528 C 1601.091553 1020.095825 1608.7854 1023.304321 1619.935303 1023.304321 C 1625.758301 1023.304321 1631.333252 1022.718018 1636.654053 1021.548828 C 1641.981201 1020.387329 1647.386475 1018.892212 1652.883057 1017.063599 L 1659.614746 1040.768066 C 1653.4646 1043.934448 1646.346191 1046.349365 1638.282959 1048.008057 C 1630.20752 1049.666748 1621.932373 1050.502441 1613.445313 1050.502441 C 1601.629639 1050.502441 1591.938965 1048.885742 1584.371582 1045.635254 C 1576.799561 1042.397095 1570.770752 1037.941528 1566.279053 1032.287231 C 1561.787842 1026.62793 1558.663574 1020.053833 1556.914551 1012.565918 C 1555.171387 1005.078003 1554.298584 997.094238 1554.298584 988.607178 C 1554.298584 981.457764 1555.12793 974.013306 1556.792969 966.276245 C 1558.45166 958.534302 1561.532471 951.464111 1566.023926 945.059692 C 1570.521484 938.655396 1576.671875 933.413818 1584.499268 929.333984 C 1592.314453 925.259033 1602.465332 923.219727 1614.946777 923.219727 C 1620.267578 923.219727 1626.211914 923.595337 1632.786133 924.345459 C 1639.360352 925.090454 1646.13916 926.712036 1653.131104 929.206421 L 1647.635498 954.17041 C 1642.30835 952.669189 1637.441162 951.712036 1633.03418 951.300415 L 1620.933105 950.670654 C 1610.952393 950.670654 1603.378906 953.879028 1598.22168 960.283447 C 1593.064453 966.687866 1590.48584 975.62854 1590.48584 987.112061 Z"/>
                    <path id="path13" d="M 1728.870117 922.717651 C 1742.010986 922.717651 1752.333008 924.80542 1759.820801 928.958374 C 1767.30249 933.116455 1772.919678 938.279785 1776.660889 944.429932 C 1780.407471 950.592529 1782.731934 957.124634 1783.651855 964.024902 C 1784.56665 970.931396 1785.020508 977.129883 1785.020508 982.620605 L 1785.020508 995.351196 L 1704.66333 995.351196 C 1704.66333 1004.914429 1707.030029 1012.130737 1711.775635 1016.996704 C 1716.515137 1021.86377 1724.378662 1024.297241 1735.358643 1024.297241 C 1742.010986 1024.297241 1748.664551 1023.595581 1755.322998 1022.178711 C 1761.975342 1020.767944 1768.464111 1018.892212 1774.790283 1016.561523 L 1782.030273 1038.395386 C 1773.876709 1042.838501 1765.305176 1045.968628 1756.321045 1047.783691 C 1747.338135 1049.593628 1738.603271 1050.502441 1730.117188 1050.502441 C 1717.804443 1050.502441 1707.671143 1048.795288 1699.729248 1045.387329 C 1691.781494 1041.97937 1685.534668 1037.366455 1680.988525 1031.537231 C 1676.44873 1025.714233 1673.307373 1019.018555 1671.569336 1011.446411 C 1669.838623 1003.874268 1668.972168 995.847046 1668.972168 987.359985 C 1668.972168 980.707764 1669.801514 973.553345 1671.466309 965.89563 C 1673.125244 958.244141 1676.211914 951.258423 1680.703613 944.93335 C 1685.195068 938.613281 1691.309326 933.328369 1699.045166 929.084839 C 1706.781982 924.842529 1716.720947 922.717651 1728.870117 922.717651 Z M 1730.249756 945.429321 C 1725.734863 945.429321 1721.96875 946.185425 1718.960938 947.680542 C 1715.952148 949.175659 1713.482666 951.216187 1711.557373 953.7948 C 1709.632324 956.379639 1708.168213 959.369873 1707.168945 962.77771 C 1706.163574 966.191895 1705.497803 969.727661 1705.159424 973.383545 L 1752.580811 973.383545 C 1752.410889 965.236084 1750.783203 958.534302 1747.684082 953.292725 C 1744.591064 948.049927 1738.779053 945.429321 1730.249756 945.429321 Z"/>
                </g>
                <g id="g3">
                    <path id="path14" d="M 268.656372 1337.826904 L 268.656372 1166.119263 L 385.708984 1166.119263 L 385.708984 1192.330566 L 305.59375 1192.330566 L 305.59375 1238.746582 L 368.238892 1238.746582 L 368.238892 1263.70459 L 305.59375 1263.70459 L 305.59375 1337.826904 L 268.656372 1337.826904 Z"/>
                    <path id="path15" d="M 517.307007 1276.435303 C 517.307007 1284.920898 516.223511 1293.032471 514.056519 1300.769409 C 511.89563 1308.505371 508.317871 1315.33374 503.330444 1321.229736 C 498.336914 1327.143311 491.847046 1331.804565 483.863281 1335.206299 C 475.873291 1338.614258 466.059814 1340.321289 454.413818 1340.321289 C 443.093994 1340.321289 433.529663 1338.662598 425.709595 1335.332642 C 417.888184 1332.010254 411.568115 1327.470337 406.744385 1321.731689 C 401.913208 1315.993164 398.463135 1309.255371 396.386597 1301.519409 C 394.305176 1293.783691 393.269897 1285.502563 393.269897 1276.683228 C 393.269897 1268.032471 394.431519 1259.836792 396.762207 1252.10083 C 399.08667 1244.365112 402.748779 1237.585083 407.742432 1231.762085 C 412.730957 1225.937866 419.17749 1221.277832 427.083252 1217.784302 C 434.982544 1214.292114 444.595337 1212.542847 455.908936 1212.542847 C 467.392456 1212.542847 477.034912 1214.249878 484.861206 1217.657837 C 492.682495 1221.072021 499.002563 1225.732056 503.826294 1231.634399 C 508.651245 1237.542847 512.10144 1244.365112 514.190308 1252.10083 C 516.266846 1259.836792 517.307007 1267.94812 517.307007 1276.435303 Z M 480.369751 1276.683228 C 480.369751 1271.858154 480.031372 1267.118774 479.371826 1262.457397 C 478.706055 1257.802246 477.501099 1253.638062 475.751831 1249.975952 C 474.002563 1246.319946 471.544189 1243.365845 468.384155 1241.119507 C 465.225464 1238.868164 461.066162 1237.748657 455.908936 1237.748657 C 450.751709 1237.748657 446.508179 1238.868164 443.178345 1241.119507 C 439.849731 1243.365845 437.191772 1246.319946 435.19458 1249.975952 C 433.197388 1253.638062 431.828735 1257.802246 431.078735 1262.457397 C 430.327515 1267.118774 429.951782 1271.858154 429.951782 1276.683228 C 429.951782 1282.173828 430.285278 1287.245361 430.95105 1291.906616 C 431.610596 1296.568237 432.906006 1300.605957 434.81897 1304.013794 C 436.731812 1307.421753 439.353882 1310.126953 442.682495 1312.125366 C 446.006104 1314.122437 450.418213 1315.115479 455.908936 1315.115479 C 460.902588 1315.115479 465.019653 1314.086548 468.263916 1311.997681 C 471.508179 1309.916138 474.002563 1307.094482 475.751831 1303.511719 C 477.501099 1299.934082 478.706055 1295.816772 479.371826 1291.156738 C 480.031372 1286.501709 480.369751 1281.676636 480.369751 1276.683228 Z"/>
                    <path id="path16" d="M 574.383423 1340.575317 C 566.229736 1340.575317 559.698975 1339.491943 554.789795 1337.331055 C 549.880493 1335.17041 546.096924 1332.173828 543.43396 1328.348145 C 540.769775 1324.522339 539.026733 1320.109009 538.197388 1315.115479 C 537.361816 1310.126953 536.94397 1304.721802 536.94397 1298.898682 L 536.94397 1215.290039 L 572.634155 1215.290039 L 573.137451 1299.642578 C 573.469727 1302.306763 574.214722 1304.637329 575.382568 1306.634644 C 576.545532 1308.626709 578.209106 1310.212402 580.376221 1311.374023 C 582.531982 1312.536743 585.528442 1313.123291 589.359131 1313.123291 C 592.852661 1313.123291 596.968506 1312.209473 601.707886 1310.376099 C 606.454834 1308.547363 611.072632 1305.635254 615.564209 1301.640991 L 615.564209 1215.290039 L 651.254272 1215.035889 L 651.254272 1337.826904 L 618.058472 1337.826904 L 617.059326 1324.600342 L 616.061279 1324.104736 C 612.235474 1327.433228 608.367676 1330.133179 604.456299 1332.216064 C 600.546265 1334.298828 596.762695 1335.956177 593.100586 1337.203369 C 589.438477 1338.456909 585.988403 1339.328125 582.744019 1339.825317 L 574.383423 1340.575317 Z"/>
                    <path id="path17" d="M 757.047729 1337.826904 L 756.545532 1253.22644 C 756.212158 1250.563477 755.420044 1248.23291 754.1604 1246.234497 C 752.906982 1244.237427 751.103271 1242.700073 748.766357 1241.616455 C 746.424561 1240.539185 743.325317 1239.993652 739.480957 1239.993652 C 735.298218 1239.993652 730.619507 1240.913574 725.431274 1242.735962 C 720.244263 1244.575684 715.552124 1246.731689 711.375488 1249.230835 L 711.375488 1337.826904 L 675.685303 1337.826904 L 675.685303 1215.290039 L 709.384399 1215.290039 L 709.874268 1226.519409 L 710.879517 1227.015381 C 718.05249 1222.191528 725.607422 1218.529419 733.530396 1216.035156 C 741.460693 1213.540771 748.34375 1212.288574 754.185181 1212.288574 C 762.362305 1212.288574 768.9552 1213.414429 773.96106 1215.659546 C 778.967163 1217.911987 782.847412 1220.944336 785.600708 1224.77002 C 788.355469 1228.602051 790.232422 1233.093384 791.236572 1238.25061 C 792.235718 1243.407837 792.737793 1248.812988 792.737793 1254.473511 L 792.737793 1337.826904 L 757.047729 1337.826904 Z"/>
                    <path id="path18" d="M 866.182129 1212.288574 C 870.177734 1212.288574 875.292847 1212.839111 881.533447 1213.928955 C 887.77417 1215.02478 893.470703 1217.082642 898.62793 1220.102661 L 898.62793 1159.382568 L 934.317993 1159.382568 L 934.317993 1337.826904 L 901.128418 1337.826904 L 900.377075 1326.853149 L 899.627075 1326.344727 C 893.640625 1331.174683 887.187866 1334.588745 880.286377 1336.579834 C 873.380005 1338.578247 867.265625 1339.576172 861.939819 1339.576172 C 852.459839 1339.576172 844.554077 1337.95459 838.229004 1334.710327 C 831.909058 1331.464844 826.915527 1327.101074 823.258423 1321.610474 C 819.596191 1316.114746 817.017578 1309.710327 815.522583 1302.384888 C 814.02124 1295.07312 813.277344 1287.331055 813.277344 1279.17749 C 813.277344 1269.697266 814.184937 1260.87793 816.019653 1252.724365 C 817.847046 1244.575684 820.885498 1237.49939 825.129028 1231.506714 C 829.372559 1225.520264 834.821167 1220.822876 841.479614 1217.408691 C 848.12561 1214.00061 856.363647 1212.288574 866.182129 1212.288574 Z M 898.62793 1245.21167 C 895.789063 1243.389282 891.939575 1241.816162 887.091309 1240.490845 C 882.241333 1239.164429 877.810669 1238.498657 873.797729 1238.498657 C 870.789063 1238.498657 867.775146 1239.037964 864.766357 1240.115234 C 861.751465 1241.192505 859.082397 1243.227051 856.739258 1246.217285 C 854.397461 1249.195068 852.508057 1253.3479 851.092407 1258.644043 C 849.669189 1263.952515 848.961304 1270.757202 848.961304 1279.043457 C 848.961304 1286.344116 849.584961 1292.276123 850.84314 1296.834717 C 852.102783 1301.398071 853.809814 1305.006836 855.989258 1307.651123 C 858.162354 1310.30896 860.752197 1312.137817 863.760986 1313.129395 C 866.769775 1314.122437 869.947144 1314.619629 873.295654 1314.619629 C 875.135376 1314.619629 877.224243 1314.334473 879.559937 1313.746826 L 886.587891 1311.634277 L 898.62793 1305.418335 L 898.62793 1245.21167 Z"/>
                    <path id="path19" d="M 1065.293701 1314.122437 L 1081.267456 1314.122437 L 1083.017944 1334.086792 C 1079.022339 1335.92041 1074.360962 1337.415527 1069.040161 1338.578247 C 1063.713135 1339.739868 1058.386108 1340.321289 1053.065186 1340.321289 C 1047.73938 1340.321289 1043.290039 1339.413818 1039.712402 1337.578979 C 1036.134399 1335.750488 1034.009644 1331.762329 1033.350098 1325.599609 L 1032.8479 1325.097534 C 1028.520142 1328.928345 1024.197266 1331.840454 1019.875488 1333.832764 C 1015.541504 1335.835938 1011.51001 1337.331055 1007.768555 1338.328979 C 1004.022217 1339.328125 1000.614136 1339.90979 997.533447 1340.073364 L 990.419922 1340.321289 C 982.768433 1340.321289 976.485596 1339.116333 971.576416 1336.70752 C 966.67334 1334.298828 962.799194 1331.302246 959.972656 1327.718262 C 957.144897 1324.146729 955.189819 1320.272705 954.112549 1316.114746 C 953.022827 1311.961548 952.484863 1308.129517 952.484863 1304.637329 C 952.484863 1298.311035 953.646362 1292.784546 955.983276 1288.038818 C 958.307739 1283.293335 961.509888 1279.303955 965.589844 1276.059692 C 969.663452 1272.815063 974.451294 1270.278809 979.942017 1268.450073 C 985.432617 1266.616699 991.419189 1265.539185 997.907837 1265.199707 L 1031.352783 1263.202637 L 1031.352783 1260.212402 C 1031.352783 1253.22644 1029.857666 1247.941528 1026.86145 1244.365112 C 1023.86499 1240.787109 1017.786743 1238.995728 1008.640137 1238.995728 C 1002.648438 1238.995728 996.116455 1239.582275 989.052612 1240.744995 C 981.976318 1241.91272 975.529785 1243.407837 969.705688 1245.236572 L 963.966919 1223.026978 C 971.286255 1219.867065 979.772095 1217.366455 989.421997 1215.539185 C 999.070679 1213.704346 1008.307861 1212.796997 1017.127197 1212.796997 C 1024.446411 1212.796997 1031.104858 1213.540771 1037.091553 1215.035889 C 1043.084229 1216.537109 1048.155884 1218.953369 1052.315186 1222.275757 C 1056.473267 1225.605713 1059.675293 1229.969482 1061.927979 1235.380737 C 1064.173096 1240.787109 1065.293701 1247.481689 1065.293701 1255.466675 L 1065.293701 1314.122437 Z M 1006.642822 1284.920898 C 995.165527 1285.914063 989.421997 1291.495117 989.421997 1301.640991 C 989.421997 1306.798218 990.75354 1310.714355 993.416382 1313.372437 C 996.074341 1316.0354 1000.317871 1317.360596 1006.146973 1317.360596 C 1009.972656 1317.360596 1014.130737 1316.404663 1018.628418 1314.491943 C 1023.119995 1312.578979 1027.3573 1309.872559 1031.352783 1306.386719 L 1031.352783 1282.923828 L 1006.642822 1284.920898 Z"/>
                    <path id="path20" d="M 1137.297485 1181.344238 L 1137.297485 1215.035889 L 1172.981445 1215.035889 L 1172.981445 1238.498657 L 1137.297485 1238.498657 L 1137.297485 1293.905273 C 1137.297485 1300.393799 1138.459106 1305.181519 1140.789795 1308.257446 C 1143.114258 1311.331909 1147.442017 1312.875244 1153.768311 1312.875244 L 1174.978516 1309.624634 L 1179.476318 1332.833496 C 1174.143066 1335.17041 1168.326294 1336.919556 1162.001221 1338.081055 C 1155.675049 1339.244019 1149.858276 1339.825317 1144.53125 1339.825317 C 1136.044189 1339.825317 1129.05835 1338.826172 1123.56897 1336.833862 C 1118.078247 1334.836792 1113.750366 1332.173828 1110.590454 1328.844116 C 1107.430298 1325.520264 1105.221069 1321.652344 1103.973999 1317.240356 C 1102.726929 1312.833008 1102.103271 1308.213989 1102.103271 1303.384033 L 1102.103271 1238.498657 L 1084.882446 1238.498657 L 1084.882446 1219.031494 L 1104.349731 1213.292847 L 1113.084717 1183.843384 L 1137.297485 1181.344238 Z"/>
                    <path id="path21" d="M 1235.626709 1181.344238 C 1235.626709 1186.670044 1233.841309 1191.167603 1230.263672 1194.824951 C 1226.686035 1198.480591 1221.407227 1200.314209 1214.41626 1200.314209 C 1207.757813 1200.314209 1202.685059 1198.480591 1199.192627 1194.824951 C 1195.699219 1191.167603 1193.949951 1186.670044 1193.949951 1181.344238 C 1193.949951 1175.853394 1195.735107 1171.282593 1199.319092 1167.620483 C 1202.890625 1163.958374 1207.926514 1162.131104 1214.41626 1162.131104 C 1221.407227 1162.131104 1226.686035 1163.958374 1230.263672 1167.620483 C 1233.841309 1171.282593 1235.626709 1175.853394 1235.626709 1181.344238 Z M 1196.698242 1337.826904 L 1196.698242 1215.290039 L 1232.382324 1215.290039 L 1232.382324 1337.826904 L 1196.698242 1337.826904 Z"/>
                    <path id="path22" d="M 1379.556152 1276.435303 C 1379.556152 1284.920898 1378.472656 1293.032471 1376.311768 1300.769409 C 1374.149658 1308.505371 1370.572998 1315.33374 1365.584473 1321.229736 C 1360.591064 1327.143311 1354.102295 1331.804565 1346.117432 1335.206299 C 1338.127441 1338.614258 1328.308838 1340.321289 1316.667969 1340.321289 C 1305.348145 1340.321289 1295.784912 1338.662598 1287.963623 1335.332642 C 1280.142334 1332.010254 1273.817383 1327.470337 1268.992188 1321.731689 C 1264.168457 1315.993164 1260.717285 1309.255371 1258.635742 1301.519409 C 1256.559082 1293.783691 1255.517822 1285.502563 1255.517822 1276.683228 C 1255.517822 1268.032471 1256.680664 1259.836792 1259.016357 1252.10083 C 1261.34082 1244.365112 1265.00293 1237.585083 1269.991455 1231.762085 C 1274.985107 1225.937866 1281.431641 1221.277832 1289.337158 1217.784302 C 1297.236816 1214.292114 1306.843018 1212.542847 1318.163086 1212.542847 C 1329.640381 1212.542847 1339.289063 1214.249878 1347.116455 1217.657837 C 1354.931641 1221.072021 1361.250488 1225.732056 1366.081787 1231.634399 C 1370.905273 1237.542847 1374.355469 1244.365112 1376.438232 1252.10083 C 1378.514648 1259.836792 1379.556152 1267.94812 1379.556152 1276.435303 Z M 1342.625 1276.683228 C 1342.625 1271.858154 1342.2854 1267.118774 1341.619629 1262.457397 C 1340.953857 1257.802246 1339.748779 1253.638062 1337.999756 1249.975952 C 1336.256592 1246.319946 1333.799561 1243.365845 1330.639404 1241.119507 C 1327.479492 1238.868164 1323.315186 1237.748657 1318.163086 1237.748657 C 1313.005859 1237.748657 1308.762207 1238.868164 1305.433594 1241.119507 C 1302.10376 1243.365845 1299.447021 1246.319946 1297.44873 1249.975952 C 1295.451416 1253.638062 1294.077881 1257.802246 1293.332764 1262.457397 C 1292.582764 1267.118774 1292.207275 1271.858154 1292.207275 1276.683228 C 1292.207275 1282.173828 1292.539307 1287.245361 1293.206299 1291.906616 C 1293.865967 1296.568237 1295.161377 1300.605957 1297.074219 1304.013794 C 1298.987061 1307.421753 1301.601807 1310.126953 1304.936523 1312.125366 C 1308.260254 1314.122437 1312.673584 1315.115479 1318.163086 1315.115479 C 1323.151611 1315.115479 1327.273682 1314.086548 1330.518066 1311.997681 C 1333.762207 1309.916138 1336.256592 1307.094482 1337.999756 1303.511719 C 1339.748779 1299.934082 1340.953857 1295.816772 1341.619629 1291.156738 C 1342.2854 1286.501709 1342.625 1281.676636 1342.625 1276.683228 Z"/>
                    <path id="path23" d="M 1480.960938 1337.826904 L 1480.45752 1253.22644 C 1480.119141 1250.563477 1479.325684 1248.23291 1478.073486 1246.234497 C 1476.820313 1244.237427 1475.016357 1242.700073 1472.679688 1241.616455 C 1470.33667 1240.539185 1467.237305 1239.993652 1463.387939 1239.993652 C 1459.211426 1239.993652 1454.531494 1240.913574 1449.344482 1242.735962 C 1444.15625 1244.575684 1439.465332 1246.731689 1435.288574 1249.230835 L 1435.288574 1337.826904 L 1399.598633 1337.826904 L 1399.598633 1215.290039 L 1433.296631 1215.290039 L 1433.787354 1226.519409 L 1434.791504 1227.015381 C 1441.965576 1222.191528 1449.519287 1218.529419 1457.443604 1216.035156 C 1465.372803 1213.540771 1472.255615 1212.288574 1478.097168 1212.288574 C 1486.275635 1212.288574 1492.867188 1213.414429 1497.873047 1215.659546 C 1502.878906 1217.911987 1506.759277 1220.944336 1509.51416 1224.77002 C 1512.268555 1228.602051 1514.144287 1233.093384 1515.149658 1238.25061 C 1516.142822 1243.407837 1516.650879 1248.812988 1516.650879 1254.473511 L 1516.650879 1337.826904 L 1480.960938 1337.826904 Z"/>
                </g>
            </g>
            <g id="g4">
                <path id="path24" d="M 1034.123657 272.993408 C 1008.632935 267.929199 981.682007 264.599609 954.509155 261.636963 L 949.974243 261.136475 C 894.325684 254.953857 848.390747 245.233154 851.924316 198.419434 C 857.227051 125.097168 1023.290527 118.03125 1023.290527 200.189941 L 1173.462646 200.189941 C 1174.674072 108.787354 1119.052856 48.713379 1045.626831 19.945557 C 904.744995 -35.260742 698.271118 24.812012 701.758179 200.189941 C 703.143433 305.777832 758.088623 356.712646 846.784912 378.389404 C 870.883179 384.28125 897.445557 388.029297 926.131104 390.107178 C 937.941772 390.777344 949.368774 391.685547 960.13269 393.006592 C 1005.670532 398.589355 1039.199951 411.628052 1039.199951 446.63562 C 1039.199951 506.784546 905.897827 516.319458 860.639038 472.830933 C 851.401001 463.942139 845.743042 452.89917 845.743042 439.569824 L 692.921875 439.569824 C 692.51416 470.391724 698.678833 497.634766 709.832642 521.389648 C 729.670044 563.680176 765.434692 594.828857 808.912598 614.998901 L 822.673584 620.883301 C 970.820313 679.834229 1192.899658 620.01062 1192.899658 447.521362 C 1192.899658 329.07373 1122.342041 290.520752 1034.123657 272.993408 Z"/>
                <path id="path25" d="M 1484.785645 151.601074 L 1667.636963 151.601074 L 1667.636963 10.259277 L 1282.491211 10.259277 L 1282.491211 628.608521 L 1443.26416 628.608521 L 1443.26416 428.090454 L 1642.011719 428.090454 L 1642.011719 286.754639 L 1443.26416 286.754639 L 1443.26416 151.601074 L 1484.785645 151.601074 Z"/>
                <path id="path26" d="M 519.792358 409.544189 C 579.70166 368.925049 610.12915 300.939941 611.258057 232.467529 C 612.748169 141.739746 562.745728 50.163086 461.400635 20.330566 C 439.647339 13.921631 415.536011 10.358154 389.055908 10.259277 L 117.864563 10.259277 L 117.864563 135.226807 C 126.67218 139.155029 134.797913 144.318359 142.08551 150.442383 L 149.728394 157.474854 C 158.047363 164.884033 167.57019 170.973633 177.978882 175.373779 C 190.261719 180.566895 203.748535 183.442383 217.917358 183.442383 C 232.074219 183.442383 245.572876 180.566895 257.837769 175.373779 C 263.677002 172.90625 269.153687 169.792236 274.375366 166.317139 L 275.988281 165.309814 C 279.498047 162.882568 282.902832 160.309814 286.087158 157.474854 L 292.427002 151.524902 C 300.536011 144.464844 309.726318 138.61377 319.779907 134.36499 C 332.056641 129.171875 345.543457 126.296631 359.706299 126.296631 C 373.875 126.296631 387.367798 129.171875 399.633911 134.36499 C 411.903564 139.550781 422.952393 147.060303 432.24292 156.338867 C 441.521484 165.612549 449.036743 176.6604 454.223877 188.925293 C 455.783203 192.62793 457.057617 196.470215 458.187744 200.375732 C 460.800903 209.433838 462.279297 218.974609 462.279297 228.869629 C 462.279297 243.03833 459.409668 256.526367 454.223877 268.813965 C 449.036743 281.090576 441.521484 292.133545 432.24292 301.400146 C 422.952393 310.684814 411.903564 318.194092 399.633911 323.386963 C 387.367798 328.579102 373.875 331.448486 359.706299 331.448486 C 347.342407 331.448486 335.555542 329.149902 324.581787 325.144287 L 319.779907 323.386963 C 309.715576 319.137207 300.518188 313.263428 292.409058 306.191406 L 286.914429 301.137939 C 283.491821 298.018066 279.800903 295.200928 275.988281 292.569824 C 270.306519 288.658447 264.27063 285.23584 257.837769 282.516602 C 251.586182 279.868652 244.99707 277.842041 238.174438 276.474609 C 231.619995 275.158447 224.849731 274.455078 217.917358 274.455078 C 203.748535 274.455078 190.261719 277.324707 177.978882 282.516602 C 167.226929 287.063477 157.419067 293.390137 148.902222 301.137939 L 145.130249 304.753662 C 137.109497 312.053223 127.894104 318.042725 117.864563 322.525146 L 117.864563 628.608521 L 275.988281 628.608521 L 275.988281 446.63562 L 358.129028 446.63562 L 464.136597 628.608521 L 655.666992 628.608521 L 519.792358 409.544189 Z"/>
            </g>
        </g>
    </g>
</svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.expand", "content.code.copy", "search", "search.suggest", "search.highlight", "navigation.footer"], "search": "../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.88dd0f4e.min.js"></script>
      
    
  </body>
</html>