---
hide:
  - toc
---
# **Introduction**

![NGS_workflow](/imgs/NGS_workflow.png)

_Typical workflow of NGS data analysis_

## **05 16S Amplicon Analysis ðŸ§«**

In the [16S amplicon analysis chapter](05_16S_amplicon_analysis) there is an introductory guide on conducting analysis using `DADA2` followed by two interesting examples of "real-life" analysis pipeline with the data from studies on Crohn's and Parkinson's diseases.

## **04 Phylogenetics ðŸŒ³**

In the [Phylogenetics chapter](04_Phylogenetics) there is a complete pipeline of simple research in phylogenetics, from working with NCBI (and other databases) to building trees, evaluating them, and getting some worthwhile results.

## **03 Whole Genome and Pangenome Analyses ðŸ§¬**

In the [Whole (pan)genome chapter](03_Whole_(pan)genome_analyses) there is a pipeline of whole genome and pangenome analyses with `PanACoTA` pipeline which includes genomes filtering with `mash`, annotating with `prokka` & `prodigal`, pangenome building with `mmseqs`, core genomes alignment with `mafft` and finally building phylogenetic tree with `iq-tree`.

## **02 Genomic Variation Analysis ðŸ”¬**

In the [Genomic Variation Analysis chapter](02_Genomic_Variation_Analysis) there is a detailed guide how to conduct studies on Variant Calling using `fastqc`, `trimmomatic`, `bwa`, `samtools`, `abra2`, `bcftools`, `snpEff` & `SnpSift`.

## **01 Quality Control of Raw Data ðŸ’Ž**

In the [Quality Control chapter](01_Quality_Control) there is a detailed guide how to conduct quality control of raw data using `fastqc` and `trimmomatic`.